// ahrs/test/test-data-random-move.js
/*********************************************************************
 *                                                                   *
 *   Copyright 2016 Simon M. Werner                                  *
 *                                                                   *
 *   Licensed to the Apache Software Foundation (ASF) under one      *
 *   or more contributor license agreements.  See the NOTICE file    *
 *   distributed with this work for additional information           *
 *   regarding copyright ownership.  The ASF licenses this file      *
 *   to you under the Apache License, Version 2.0 (the               *
 *   "License"); you may not use this file except in compliance      *
 *   with the License.  You may obtain a copy of the License at      *
 *                                                                   *
 *      http://www.apache.org/licenses/LICENSE-2.0                   *
 *                                                                   *
 *   Unless required by applicable law or agreed to in writing,      *
 *   software distributed under the License is distributed on an     *
 *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY          *
 *   KIND, either express or implied.  See the License for the       *
 *   specific language governing permissions and limitations         *
 *   under the License.                                              *
 *                                                                   *
 *********************************************************************/

'use strict';

//
// This data was gathered using the mpu9250 library (https://www.npmjs.com/package/mpu9250)
// and using the `create-test-data.js` script.
//

module.exports = [{
    gyro: {
        x: -0.12098,
        y: -0.39006,
        z: -0.25147
    },
    accel: {
        x: -0.00587,
        y: -0.9991,
        z: 0.00777
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 36.85401
    },
    dt: 29
}, {
    gyro: {
        x: -0.15151,
        y: -0.25266,
        z: 0.15311
    },
    accel: {
        x: -0.00172,
        y: -0.99836,
        z: 0.00849
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 33.42573
    },
    dt: 54
}, {
    gyro: {
        x: -0.05991,
        y: -0.4206,
        z: -0.00719
    },
    accel: {
        x: -0.0044,
        y: -1.0013,
        z: 0.01136
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 33.42573
    },
    dt: 19
}, {
    gyro: {
        x: 0.0546,
        y: -0.09998,
        z: -0.09116
    },
    accel: {
        x: -0.00074,
        y: -1.00032,
        z: 0.00609
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 15
}, {
    gyro: {
        x: 0.01643,
        y: -0.61907,
        z: 0.13785
    },
    accel: {
        x: -0.00269,
        y: -1.00277,
        z: 0.01064
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 17
}, {
    gyro: {
        x: 0.02406,
        y: -0.32136,
        z: -0.06063
    },
    accel: {
        x: -0.00416,
        y: -1.00203,
        z: 0.0128
    },
    compass: {
        x: -125.02294,
        y: -37.63317,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 0.00879,
        y: -0.01602,
        z: 0.16838
    },
    accel: {
        x: -0.00221,
        y: -1.0013,
        z: 0.01304
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 38.56815
    },
    dt: 19
}, {
    gyro: {
        x: -0.16678,
        y: -0.09998,
        z: 0.11495
    },
    accel: {
        x: -0.00294,
        y: -1.00448,
        z: 0.014
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 38.56815
    },
    dt: 19
}, {
    gyro: {
        x: 0.25307,
        y: -0.37479,
        z: -0.03009
    },
    accel: {
        x: -0.00856,
        y: -1.00228,
        z: 0.01088
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 38.56815
    },
    dt: 16
}, {
    gyro: {
        x: -0.13624,
        y: -0.14579,
        z: -0.09879
    },
    accel: {
        x: -0.00489,
        y: -1.0062,
        z: 0.01112
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 36.85401
    },
    dt: 16
}, {
    gyro: {
        x: 0.06986,
        y: -0.30609,
        z: 0.16075
    },
    accel: {
        x: -0.00147,
        y: -0.99934,
        z: 0.00968
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 36.85401
    },
    dt: 17
}, {
    gyro: {
        x: -0.01411,
        y: -0.19922,
        z: -0.1675
    },
    accel: {
        x: -0.00416,
        y: -0.99812,
        z: 0.0164
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 35.13987
    },
    dt: 25
}, {
    gyro: {
        x: -0.06754,
        y: -0.06945,
        z: -0.02246
    },
    accel: {
        x: -0.00465,
        y: -1.00399,
        z: 0.0164
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 35.13987
    },
    dt: 16
}, {
    gyro: {
        x: 0.08513,
        y: -0.09998,
        z: -0.07589
    },
    accel: {
        x: -0.00391,
        y: -0.99983,
        z: 0.01472
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: 0.10803,
        y: -0.45113,
        z: 0.03861
    },
    accel: {
        x: -0.00245,
        y: -1.00277,
        z: 0.00729
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 35.13987
    },
    dt: 20
}, {
    gyro: {
        x: -0.02937,
        y: -0.51983,
        z: 0.15311
    },
    accel: {
        x: -0.00294,
        y: -1.00155,
        z: 0.01424
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 35.13987
    },
    dt: 16
}, {
    gyro: {
        x: -0.15151,
        y: -0.11525,
        z: 0.21418
    },
    accel: {
        x: -0.00636,
        y: -0.99861,
        z: 0.0152
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 36.85401
    },
    dt: 16
}, {
    gyro: {
        x: -0.16678,
        y: -0.39006,
        z: 0.00808
    },
    accel: {
        x: -0.00318,
        y: -1.00106,
        z: 0.01016
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 18
}, {
    gyro: {
        x: 0.1233,
        y: -0.17632,
        z: 0.23708
    },
    accel: {
        x: -0.00465,
        y: -1.00375,
        z: 0.00512
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 17
}, {
    gyro: {
        x: -0.01411,
        y: -0.24502,
        z: 0.11495
    },
    accel: {
        x: -0.00758,
        y: -1.00497,
        z: 0.01112
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 18
}, {
    gyro: {
        x: -0.02937,
        y: -0.47403,
        z: 0.03861
    },
    accel: {
        x: -0.00416,
        y: -1.00277,
        z: 0.0128
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 14
}, {
    gyro: {
        x: -0.11334,
        y: -0.32136,
        z: 0.03098
    },
    accel: {
        x: -0.00221,
        y: -1.00203,
        z: 0.014
    },
    compass: {
        x: -123.17075,
        y: -43.00933,
        z: 36.85401
    },
    dt: 18
}, {
    gyro: {
        x: 0.2149,
        y: -0.31373,
        z: 0.03098
    },
    accel: {
        x: -0.00221,
        y: -1.00155,
        z: 0.00753
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 35.13987
    },
    dt: 14
}, {
    gyro: {
        x: 0.00879,
        y: -0.15342,
        z: -0.02246
    },
    accel: {
        x: -0.00465,
        y: -1.00424,
        z: 0.01304
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 40.28229
    },
    dt: 13
}, {
    gyro: {
        x: -0.05991,
        y: -0.35953,
        z: -0.01482
    },
    accel: {
        x: -0.00196,
        y: -1.00203,
        z: 0.01184
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 40.28229
    },
    dt: 14
}, {
    gyro: {
        x: -0.08281,
        y: -0.36716,
        z: 0.16838
    },
    accel: {
        x: 0.00073,
        y: -1.00081,
        z: 0.01328
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 40.28229
    },
    dt: 21
}, {
    gyro: {
        x: -0.05227,
        y: -0.30609,
        z: 0.05388
    },
    accel: {
        x: -0.00172,
        y: -1.00155,
        z: 0.01352
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 14
}, {
    gyro: {
        x: -0.21258,
        y: -0.13052,
        z: 0.13785
    },
    accel: {
        x: -0.00538,
        y: -1.00179,
        z: 0.02193
    },
    compass: {
        x: -126.87513,
        y: -46.59345,
        z: 38.56815
    },
    dt: 26
}, {
    gyro: {
        x: -0.22021,
        y: -0.26029,
        z: 0.03861
    },
    accel: {
        x: -0.00245,
        y: -1.00424,
        z: 0.01184
    },
    compass: {
        x: -126.87513,
        y: -46.59345,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: -0.13624,
        y: -0.31373,
        z: -0.06826
    },
    accel: {
        x: -0.0044,
        y: -1.00155,
        z: 0.00873
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: -0.19731,
        y: -0.22976,
        z: 0.29052
    },
    accel: {
        x: -0.0005,
        y: -1.00448,
        z: 0.01688
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 36.85401
    },
    dt: 14
}, {
    gyro: {
        x: 0.06223,
        y: -0.32899,
        z: 0.06151
    },
    accel: {
        x: -0.00098,
        y: -1.00155,
        z: 0.01496
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 35.13987
    },
    dt: 62
}, {
    gyro: {
        x: -0.12098,
        y: -0.06182,
        z: 0.23708
    },
    accel: {
        x: -0.00782,
        y: -1.00252,
        z: 0.00657
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 36.85401
    },
    dt: 15
}, {
    gyro: {
        x: -0.17441,
        y: 0.04505,
        z: 0.16075
    },
    accel: {
        x: -0.0066,
        y: -0.99885,
        z: 0.00801
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 35.13987
    },
    dt: 23
}, {
    gyro: {
        x: -0.16678,
        y: -0.31373,
        z: -0.06826
    },
    accel: {
        x: -0.00514,
        y: -1.00081,
        z: 0.01208
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: -0.05227,
        y: -0.35189,
        z: 0.03098
    },
    accel: {
        x: -0.00391,
        y: -0.99983,
        z: 0.00873
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 36.85401
    },
    dt: 19
}, {
    gyro: {
        x: -0.08281,
        y: -0.06945,
        z: 0.09968
    },
    accel: {
        x: -0.00245,
        y: -0.99934,
        z: 0.01064
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.08281,
        y: -0.47403,
        z: 0.09205
    },
    accel: {
        x: -0.00123,
        y: -1.00081,
        z: 0.00944
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: -0.25838,
        y: -0.48166,
        z: 0.10731
    },
    accel: {
        x: -0.00343,
        y: -1.00375,
        z: 0.00392
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 36.85401
    },
    dt: 14
}, {
    gyro: {
        x: 0.18437,
        y: -0.09235,
        z: 0.17602
    },
    accel: {
        x: -0.00221,
        y: -1.00448,
        z: 0.00705
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.07518,
        y: -0.09235,
        z: 0.06151
    },
    accel: {
        x: -0.0066,
        y: -1.00203,
        z: 0.0116
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: 0.0775,
        y: -0.26792,
        z: 0.01571
    },
    accel: {
        x: -0.00489,
        y: -1.00326,
        z: 0.01328
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 36.85401
    },
    dt: 14
}, {
    gyro: {
        x: 0.03169,
        y: -0.37479,
        z: 0.15311
    },
    accel: {
        x: -0.00294,
        y: -1.00106,
        z: 0.01088
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -0.01411,
        y: -0.39769,
        z: 0.10731
    },
    accel: {
        x: 0.00073,
        y: -1.00155,
        z: 0.00609
    },
    compass: {
        x: -121.31856,
        y: -43.00933,
        z: 36.85401
    },
    dt: 19
}, {
    gyro: {
        x: 0.06986,
        y: -0.09235,
        z: 0.05388
    },
    accel: {
        x: -0.0044,
        y: -1.00473,
        z: 0.01424
    },
    compass: {
        x: -121.31856,
        y: -43.00933,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.07518,
        y: -0.26029,
        z: 0.19128
    },
    accel: {
        x: -0.00587,
        y: -0.99689,
        z: 0.01208
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 19
}, {
    gyro: {
        x: 0.00879,
        y: -0.26029,
        z: 0.23708
    },
    accel: {
        x: -0.00294,
        y: -1.00155,
        z: 0.01688
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 14
}, {
    gyro: {
        x: -0.21258,
        y: -0.54273,
        z: 0.02334
    },
    accel: {
        x: -0.00001,
        y: -1.00424,
        z: 0.01016
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.03933,
        y: -0.39006,
        z: -0.18276
    },
    accel: {
        x: -0.00196,
        y: -1.00448,
        z: 0.00729
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: 0.15383,
        y: -0.16869,
        z: 0.00044
    },
    accel: {
        x: -0.00221,
        y: -1.00448,
        z: 0.0152
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 38.56815
    },
    dt: 14
}, {
    gyro: {
        x: 0.04696,
        y: -0.01602,
        z: -0.18276
    },
    accel: {
        x: -0.00196,
        y: -1.00032,
        z: 0.01064
    },
    compass: {
        x: -123.17075,
        y: -39.42522,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 0.13856,
        y: -0.41296,
        z: 0.08441
    },
    accel: {
        x: -0.00318,
        y: -1.00301,
        z: 0.00992
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.01411,
        y: -0.24502,
        z: 0.05388
    },
    accel: {
        x: 0.00366,
        y: -1.00081,
        z: 0.01112
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 36.85401
    },
    dt: 14
}, {
    gyro: {
        x: -0.12098,
        y: -0.29082,
        z: 0.03861
    },
    accel: {
        x: -0.00294,
        y: -0.99885,
        z: 0.00392
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.12098,
        y: -0.17632,
        z: 0.18365
    },
    accel: {
        x: -0.00172,
        y: -1.0013,
        z: 0.00633
    },
    compass: {
        x: -123.17075,
        y: -46.59345,
        z: 40.28229
    },
    dt: 12
}, {
    gyro: {
        x: -0.08281,
        y: -0.02365,
        z: -0.02246
    },
    accel: {
        x: -0.00514,
        y: -0.99812,
        z: 0.01256
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.22021,
        y: -0.28319,
        z: -0.15223
    },
    accel: {
        x: -0.0044,
        y: -1.00203,
        z: 0.00992
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 36.85401
    },
    dt: 14
}, {
    gyro: {
        x: 0.00879,
        y: -0.11525,
        z: 0.26762
    },
    accel: {
        x: -0.00123,
        y: -0.99836,
        z: 0.00657
    },
    compass: {
        x: -130.57951,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.00116,
        y: -0.15342,
        z: 0.00808
    },
    accel: {
        x: -0.00172,
        y: -1.00057,
        z: 0.0116
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 0.0775,
        y: -0.23739,
        z: 0.25235
    },
    accel: {
        x: -0.00123,
        y: -1.00106,
        z: 0.00681
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 35.13987
    },
    dt: 14
}, {
    gyro: {
        x: 0.22253,
        y: -0.4206,
        z: -0.02246
    },
    accel: {
        x: -0.00172,
        y: -0.99861,
        z: 0.01256
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.1233,
        y: -0.26029,
        z: -0.06826
    },
    accel: {
        x: -0.00514,
        y: -0.99665,
        z: 0.01616
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.15383,
        y: -0.29846,
        z: -0.10643
    },
    accel: {
        x: -0.00538,
        y: -1.0013,
        z: 0.01448
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: -0.07518,
        y: -0.00075,
        z: 0.08441
    },
    accel: {
        x: -0.00367,
        y: -0.99689,
        z: 0.01448
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 38.56815
    },
    dt: 14
}, {
    gyro: {
        x: 0.34467,
        y: -0.19159,
        z: 0.00808
    },
    accel: {
        x: -0.00416,
        y: -1.0035,
        z: 0.00633
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -0.07518,
        y: 0.00689,
        z: 0.06915
    },
    accel: {
        x: -0.00367,
        y: -0.99934,
        z: 0.00992
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 33
}, {
    gyro: {
        x: -0.02937,
        y: -0.11525,
        z: 0.10731
    },
    accel: {
        x: -0.00562,
        y: -0.99861,
        z: 0.01472
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 36.85401
    },
    dt: 14
}, {
    gyro: {
        x: 0.08513,
        y: -0.07708,
        z: 0.00044
    },
    accel: {
        x: -0.00294,
        y: -1.00179,
        z: 0.01328
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.00879,
        y: 0.00689,
        z: -0.19803
    },
    accel: {
        x: -0.00416,
        y: -1.00008,
        z: -0.01076
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 0.08513,
        y: -0.23739,
        z: 0.17602
    },
    accel: {
        x: -0.00587,
        y: -1.00693,
        z: 0.01304
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 35.13987
    },
    dt: 14
}, {
    gyro: {
        x: 0.2378,
        y: 0.04505,
        z: -0.07589
    },
    accel: {
        x: -0.00562,
        y: -1.0035,
        z: 0.00968
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 38.56815
    },
    dt: 26
}, {
    gyro: {
        x: 0.00879,
        y: -0.11525,
        z: 0.00808
    },
    accel: {
        x: -0.00343,
        y: -1.00081,
        z: 0.00992
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 40.28229
    },
    dt: 13
}, {
    gyro: {
        x: -0.00647,
        y: 0.15956,
        z: 0.13785
    },
    accel: {
        x: -0.00709,
        y: -1.00155,
        z: 0.01064
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 40.28229
    },
    dt: 14
}, {
    gyro: {
        x: -0.02937,
        y: -0.16105,
        z: 0.02334
    },
    accel: {
        x: -0.00318,
        y: -0.99983,
        z: 0.01016
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: 0.13093,
        y: 0.02215,
        z: -0.1675
    },
    accel: {
        x: -0.00416,
        y: -1.00522,
        z: 0.00657
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 31.71159
    },
    dt: 14
}, {
    gyro: {
        x: 0.08513,
        y: -0.09235,
        z: 0.13785
    },
    accel: {
        x: 0.00048,
        y: -1.00252,
        z: 0.00705
    },
    compass: {
        x: -121.31856,
        y: -37.63317,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: 0.02406,
        y: 0.09849,
        z: 0.06151
    },
    accel: {
        x: -0.00416,
        y: -0.99641,
        z: 0.01136
    },
    compass: {
        x: -121.31856,
        y: -37.63317,
        z: 33.42573
    },
    dt: 12
}, {
    gyro: {
        x: 0.23017,
        y: 0.27406,
        z: 0.19892
    },
    accel: {
        x: -0.00147,
        y: -1.00081,
        z: 0.01833
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.12861,
        y: -0.08472,
        z: -0.11406
    },
    accel: {
        x: 0.00292,
        y: -1.00473,
        z: 0.0128
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 0.1004,
        y: -0.16105,
        z: 0.16075
    },
    accel: {
        x: -0.00221,
        y: -1.00252,
        z: 0.0104
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 36.85401
    },
    dt: 14
}, {
    gyro: {
        x: 0.3065,
        y: -0.32899,
        z: -0.13696
    },
    accel: {
        x: -0.00001,
        y: -1.00252,
        z: 0.00536
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.00116,
        y: 0.11376,
        z: -0.03009
    },
    accel: {
        x: -0.00221,
        y: -1.0013,
        z: 0.0152
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.48208,
        y: -0.07708,
        z: -0.20566
    },
    accel: {
        x: -0.00074,
        y: -1.00277,
        z: 0.01376
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.19963,
        y: 0.05269,
        z: -0.33544
    },
    accel: {
        x: -0.00514,
        y: -1.00252,
        z: 0.00488
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 14
}, {
    gyro: {
        x: 0.38284,
        y: -0.01602,
        z: -0.17513
    },
    accel: {
        x: -0.00147,
        y: -0.99861,
        z: 0.0116
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: 0.11566,
        y: -0.15342,
        z: -0.13696
    },
    accel: {
        x: -0.00367,
        y: -1.00301,
        z: 0.0128
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.3294,
        y: 0.16719,
        z: -0.28963
    },
    accel: {
        x: -0.00098,
        y: -1.00203,
        z: 0.01232
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: 0.55841,
        y: 0.27406,
        z: -0.3049
    },
    accel: {
        x: -0.00196,
        y: -1.00301,
        z: 0.02001
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 38.56815
    },
    dt: 14
}, {
    gyro: {
        x: 0.70345,
        y: 0.58704,
        z: -0.2133
    },
    accel: {
        x: -0.00221,
        y: -1.00644,
        z: 0.02001
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.50498,
        y: 0.31223,
        z: -0.1446
    },
    accel: {
        x: -0.00098,
        y: -1.00008,
        z: 0.00633
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 47
}, {
    gyro: {
        x: 0.43627,
        y: 0.06795,
        z: -0.51864
    },
    accel: {
        x: 0.00146,
        y: -0.99714,
        z: 0.01448
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 40.28229
    },
    dt: 14
}, {
    gyro: {
        x: 2.55078,
        y: 0.66337,
        z: -20.3736
    },
    accel: {
        x: 0.04029,
        y: -1.17776,
        z: -0.19484
    },
    compass: {
        x: -130.57951,
        y: -39.42522,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -10.49502,
        y: -8.98548,
        z: -28.40414
    },
    accel: {
        x: 0.01489,
        y: -1.0248,
        z: -0.03458
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 3.00116,
        y: 1.04505,
        z: -24.67895
    },
    accel: {
        x: 0.02051,
        y: -1.00301,
        z: -0.01317
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 0.65002,
        y: -1.03128,
        z: -21.51101
    },
    accel: {
        x: 0.03492,
        y: -0.95602,
        z: 0.02914
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 38.56815
    },
    dt: 14
}, {
    gyro: {
        x: 2.70345,
        y: 1.10612,
        z: -20.2591
    },
    accel: {
        x: -0.01344,
        y: -1.05417,
        z: 0.00392
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 1.48208,
        y: 0.44963,
        z: -11.77055
    },
    accel: {
        x: -0.01778,
        y: -1.09186,
        z: 0.0032
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 2.76452,
        y: 2.28169,
        z: -7.33544
    },
    accel: {
        x: 0.03492,
        y: -0.99592,
        z: 0.04883
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 3.29124,
        y: 1.5336,
        z: -6.40414
    },
    accel: {
        x: 0.03932,
        y: -1.01476,
        z: -0.00258
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 9.48208,
        y: -3.16105,
        z: 0.93937
    },
    accel: {
        x: 0.02417,
        y: -1.12906,
        z: 0.00968
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 36.85401
    },
    dt: 19
}, {
    gyro: {
        x: 12.56605,
        y: -10.32899,
        z: 5.13021
    },
    accel: {
        x: 0.04445,
        y: -1.19759,
        z: -0.04565
    },
    compass: {
        x: -126.87513,
        y: -35.84111,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 14.40574,
        y: -17.51983,
        z: 7.27525
    },
    accel: {
        x: 0.0696,
        y: -1.09577,
        z: -0.1046
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 15.1233,
        y: 0.52597,
        z: 10.06915
    },
    accel: {
        x: 0.03174,
        y: -1.03287,
        z: 0.12066
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 18.79505,
        y: -11.21449,
        z: 16.74853
    },
    accel: {
        x: -0.00562,
        y: -0.92078,
        z: 0.02601
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 23.65765,
        y: -3.79464,
        z: 14.21418
    },
    accel: {
        x: 0.02222,
        y: -0.90511,
        z: 0.08199
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 31.71159
    },
    dt: 12
}, {
    gyro: {
        x: 22.04696,
        y: 3.36566,
        z: 9.87067
    },
    accel: {
        x: 0.01245,
        y: -0.95822,
        z: 0.04115
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: 14.43627,
        y: 7.99925,
        z: 3.99281
    },
    accel: {
        x: 0.03687,
        y: -0.86375,
        z: 0.13363
    },
    compass: {
        x: -130.57951,
        y: -41.21728,
        z: 35.13987
    },
    dt: 21
}, {
    gyro: {
        x: 4.22253,
        y: -0.4206,
        z: -3.74002
    },
    accel: {
        x: 0.04762,
        y: -0.87501,
        z: 0.07742
    },
    compass: {
        x: -130.57951,
        y: -41.21728,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.46681,
        y: -1.74884,
        z: -16.04536
    },
    accel: {
        x: 0.0508,
        y: -0.92176,
        z: -0.01076
    },
    compass: {
        x: -130.57951,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -9.38052,
        y: -6.29082,
        z: -23.51101
    },
    accel: {
        x: 0.03639,
        y: -0.94746,
        z: -0.01196
    },
    compass: {
        x: -128.72732,
        y: -37.63317,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: -16.52556,
        y: -15.87098,
        z: -24.09879
    },
    accel: {
        x: 0.0083,
        y: -0.96165,
        z: -0.01124
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 28.28331
    },
    dt: 14
}, {
    gyro: {
        x: -9.808,
        y: -23.4893,
        z: -20.43467
    },
    accel: {
        x: -0.00367,
        y: -0.99983,
        z: 0.02962
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 28.28331
    },
    dt: 13
}, {
    gyro: {
        x: -8.09808,
        y: -22.71831,
        z: -16.19803
    },
    accel: {
        x: -0.02216,
        y: -1.00277,
        z: 0.00175
    },
    compass: {
        x: -130.57951,
        y: -39.42522,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 2.48971,
        y: -22.87861,
        z: -7.54918
    },
    accel: {
        x: -0.04213,
        y: -1.10532,
        z: 0.08991
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 9.58131,
        y: -24.22976,
        z: 1.87831
    },
    accel: {
        x: 0.00317,
        y: -1.15696,
        z: 0.03754
    },
    compass: {
        x: -128.72732,
        y: -34.04906,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 20.61185,
        y: -21.66487,
        z: 5.51953
    },
    accel: {
        x: 0.07742,
        y: -1.155,
        z: 0.04835
    },
    compass: {
        x: -128.72732,
        y: -34.04906,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 15.03169,
        y: -17.76411,
        z: 4.40502
    },
    accel: {
        x: 0.0972,
        y: -1.0385,
        z: 0.10793
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 31.71159
    },
    dt: 12
}, {
    gyro: {
        x: 2.93246,
        y: -7.90151,
        z: -2.49574
    },
    accel: {
        x: 0.07156,
        y: -0.98441,
        z: 0.13411
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: -6.44159,
        y: -1.06182,
        z: -6.48047
    },
    accel: {
        x: 0.01489,
        y: -1.02161,
        z: 0.11538
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 28.28331
    },
    dt: 13
}, {
    gyro: {
        x: 4.36757,
        y: 2.24353,
        z: -4.24383
    },
    accel: {
        x: 0.021,
        y: -1.05563,
        z: 0.13628
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 28.28331
    },
    dt: 13
}, {
    gyro: {
        x: 7.3294,
        y: 8.86185,
        z: 1.48899
    },
    accel: {
        x: 0.00317,
        y: -1.07056,
        z: 0.13796
    },
    compass: {
        x: -132.43171,
        y: -35.84111,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: 12.33704,
        y: 14.86948,
        z: 5.09205
    },
    accel: {
        x: 0.01831,
        y: -1.03508,
        z: 0.14324
    },
    compass: {
        x: -134.2839,
        y: -37.63317,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: 15.2607,
        y: 20.11376,
        z: 7.65693
    },
    accel: {
        x: 0.02393,
        y: -1.05368,
        z: 0.13483
    },
    compass: {
        x: -132.43171,
        y: -35.84111,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: 18.77979,
        y: 23.24353,
        z: 3.96227
    },
    accel: {
        x: 0.05031,
        y: -1.02749,
        z: 0.12402
    },
    compass: {
        x: -132.43171,
        y: -35.84111,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: 7.78742,
        y: 24.94582,
        z: 2.35922
    },
    accel: {
        x: 0.02588,
        y: -1.09308,
        z: 0.07718
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: 7.55841,
        y: 23.09085,
        z: 3.21418
    },
    accel: {
        x: 0.00488,
        y: -1.08574,
        z: 0.07886
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: 8.66528,
        y: 21.70154,
        z: 2.55006
    },
    accel: {
        x: -0.00807,
        y: -1.0341,
        z: 0.11009
    },
    compass: {
        x: -130.57951,
        y: -37.63317,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: 7.87902,
        y: 24.60994,
        z: 2.77907
    },
    accel: {
        x: -0.00587,
        y: -1.03043,
        z: 0.14925
    },
    compass: {
        x: -130.57951,
        y: -37.63317,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: 16.0546,
        y: 29.60231,
        z: 6.14548
    },
    accel: {
        x: 0.0061,
        y: -1.0571,
        z: 0.15021
    },
    compass: {
        x: -132.43171,
        y: -39.42522,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: 23.35231,
        y: 28.85421,
        z: 12.80197
    },
    accel: {
        x: -0.02436,
        y: -1.05857,
        z: 0.14324
    },
    compass: {
        x: -128.72732,
        y: -37.63317,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: 33.23017,
        y: 23.96872,
        z: 18.15311
    },
    accel: {
        x: -0.01393,
        y: -1.09112,
        z: 0.18552
    },
    compass: {
        x: -130.57951,
        y: -37.63317,
        z: 29.99745
    },
    dt: 15
}, {
    gyro: {
        x: 21.54315,
        y: 44.72444,
        z: 18.18365
    },
    accel: {
        x: -0.03069,
        y: -1.05955,
        z: 0.22684
    },
    compass: {
        x: -130.57951,
        y: -41.21728,
        z: 23.14089
    },
    dt: 96
}, {
    gyro: {
        x: 31.93246,
        y: 34.57177,
        z: 22.88594
    },
    accel: {
        x: -0.07062,
        y: -1.10116,
        z: 0.2708
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 23.14089
    },
    dt: 13
}, {
    gyro: {
        x: 42.03933,
        y: 45.31986,
        z: 24.36685
    },
    accel: {
        x: -0.00318,
        y: -1.00889,
        z: 0.34815
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 24.85503
    },
    dt: 16
}, {
    gyro: {
        x: 55.58895,
        y: 50.49544,
        z: 23.80197
    },
    accel: {
        x: -0.07427,
        y: -1.02284,
        z: 0.33134
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 21.42675
    },
    dt: 13
}, {
    gyro: {
        x: 73.65002,
        y: 53.3504,
        z: 32.16075
    },
    accel: {
        x: -0.10325,
        y: -1.11486,
        z: 0.34095
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 21.42675
    },
    dt: 12
}, {
    gyro: {
        x: 99.13856,
        y: 56.11376,
        z: 39.43556
    },
    accel: {
        x: -0.01827,
        y: -1.08158,
        z: 0.37506
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 21.42675
    },
    dt: 13
}, {
    gyro: {
        x: 103.86376,
        y: 62.5107,
        z: 37.09205
    },
    accel: {
        x: -0.02241,
        y: -1.09969,
        z: 0.36377
    },
    compass: {
        x: -130.57951,
        y: -50.17756,
        z: 14.57019
    },
    dt: 13
}, {
    gyro: {
        x: 95.42864,
        y: 64.12139,
        z: 42.36685
    },
    accel: {
        x: -0.05844,
        y: -1.20836,
        z: 0.35488
    },
    compass: {
        x: -128.72732,
        y: -51.96961,
        z: 12.85605
    },
    dt: 12
}, {
    gyro: {
        x: 97.99353,
        y: 61.3733,
        z: 57.42029
    },
    accel: {
        x: -0.11079,
        y: -1.24531,
        z: 0.38683
    },
    compass: {
        x: -126.87513,
        y: -48.3855,
        z: 14.57019
    },
    dt: 12
}, {
    gyro: {
        x: 110.52788,
        y: 59.04505,
        z: 72.24472
    },
    accel: {
        x: -0.15779,
        y: -1.14619,
        z: 0.43824
    },
    compass: {
        x: -126.87513,
        y: -48.3855,
        z: 14.57019
    },
    dt: 13
}, {
    gyro: {
        x: 137.2836,
        y: 60.77024,
        z: 74.65693
    },
    accel: {
        x: -0.12297,
        y: -0.93277,
        z: 0.56411
    },
    compass: {
        x: -128.72732,
        y: -55.55372,
        z: 7.71363
    },
    dt: 13
}, {
    gyro: {
        x: 157.51261,
        y: 69.95345,
        z: 73.25998
    },
    accel: {
        x: -0.15608,
        y: -0.96288,
        z: 0.65732
    },
    compass: {
        x: -126.87513,
        y: -53.76167,
        z: 5.99949
    },
    dt: 13
}, {
    gyro: {
        x: 173.33704,
        y: 82.23589,
        z: 74.59586
    },
    accel: {
        x: -0.18165,
        y: -1.00522,
        z: 0.69984
    },
    compass: {
        x: -130.57951,
        y: -55.55372,
        z: 2.57121
    },
    dt: 12
}, {
    gyro: {
        x: 173.69582,
        y: 100.65574,
        z: 70.50426
    },
    accel: {
        x: -0.14026,
        y: -0.89263,
        z: 0.69455
    },
    compass: {
        x: -130.57951,
        y: -55.55372,
        z: 2.57121
    },
    dt: 13
}, {
    gyro: {
        x: 160.76452,
        y: 112.73208,
        z: 61.04624
    },
    accel: {
        x: -0.18822,
        y: -0.8427,
        z: 0.63186
    },
    compass: {
        x: -123.17075,
        y: -60.92989,
        z: -4.28535
    },
    dt: 12
}, {
    gyro: {
        x: 150.59658,
        y: 120.70154,
        z: 50.33632
    },
    accel: {
        x: -0.18189,
        y: -0.81456,
        z: 0.5963
    },
    compass: {
        x: -125.02294,
        y: -62.72195,
        z: -7.71363
    },
    dt: 12
}, {
    gyro: {
        x: 134.3065,
        y: 122.27406,
        z: 41.95464
    },
    accel: {
        x: -0.1405,
        y: -0.81529,
        z: 0.56363
    },
    compass: {
        x: -123.17075,
        y: -64.514,
        z: -9.42777
    },
    dt: 13
}, {
    gyro: {
        x: 112.31414,
        y: 118.66337,
        z: 43.88594
    },
    accel: {
        x: -0.20649,
        y: -0.90438,
        z: 0.5276
    },
    compass: {
        x: -123.17075,
        y: -64.514,
        z: -9.42777
    },
    dt: 13
}, {
    gyro: {
        x: 109.0775,
        y: 111.84658,
        z: 47.30579
    },
    accel: {
        x: -0.24763,
        y: -0.78788,
        z: 0.55066
    },
    compass: {
        x: -123.17075,
        y: -64.514,
        z: -9.42777
    },
    dt: 12
}, {
    gyro: {
        x: 120.192,
        y: 108.34276,
        z: 42.48136
    },
    accel: {
        x: -0.26224,
        y: -0.70858,
        z: 0.55378
    },
    compass: {
        x: -119.46636,
        y: -69.89017,
        z: -14.57019
    },
    dt: 13
}, {
    gyro: {
        x: 131.59658,
        y: 107.92292,
        z: 38.88594
    },
    accel: {
        x: -0.28465,
        y: -0.70075,
        z: 0.59222
    },
    compass: {
        x: -121.31856,
        y: -69.89017,
        z: -19.71261
    },
    dt: 13
}, {
    gyro: {
        x: 150.22253,
        y: 110.20536,
        z: 42.34395
    },
    accel: {
        x: -0.33334,
        y: -0.78886,
        z: 0.60783
    },
    compass: {
        x: -121.31856,
        y: -69.89017,
        z: -19.71261
    },
    dt: 13
}, {
    gyro: {
        x: 161.3065,
        y: 113.42673,
        z: 44.94701
    },
    accel: {
        x: -0.32263,
        y: -0.72694,
        z: 0.60303
    },
    compass: {
        x: -121.31856,
        y: -69.89017,
        z: -19.71261
    },
    dt: 12
}, {
    gyro: {
        x: 163.89429,
        y: 114.58704,
        z: 42.07678
    },
    accel: {
        x: -0.31752,
        y: -0.59551,
        z: 0.63786
    },
    compass: {
        x: -117.61417,
        y: -75.26634,
        z: -17.99847
    },
    dt: 12
}, {
    gyro: {
        x: 187.75689,
        y: 112.69391,
        z: 28.29052
    },
    accel: {
        x: -0.33626,
        y: -0.46995,
        z: 0.73779
    },
    compass: {
        x: -113.90979,
        y: -73.47428,
        z: -21.42675
    },
    dt: 13
}, {
    gyro: {
        x: 195.59658,
        y: 122.36566,
        z: 22.8096
    },
    accel: {
        x: -0.34552,
        y: -0.77099,
        z: 0.75004
    },
    compass: {
        x: -112.0576,
        y: -75.26634,
        z: -23.14089
    },
    dt: 13
}, {
    gyro: {
        x: 194.80269,
        y: 130.00689,
        z: 28.64166
    },
    accel: {
        x: -0.33797,
        y: -0.84026,
        z: 0.75101
    },
    compass: {
        x: -112.0576,
        y: -75.26634,
        z: -23.14089
    },
    dt: 13
}, {
    gyro: {
        x: 185.84085,
        y: 137.60994,
        z: 34.37449
    },
    accel: {
        x: -0.37011,
        y: -0.74358,
        z: 0.77359
    },
    compass: {
        x: -106.50102,
        y: -77.05839,
        z: -31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 196.97826,
        y: 143.13666,
        z: 30.21418
    },
    accel: {
        x: -0.38253,
        y: -0.55072,
        z: 0.78896
    },
    compass: {
        x: -106.50102,
        y: -80.6425,
        z: -36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 204.36757,
        y: 145.96108,
        z: 23.6493
    },
    accel: {
        x: -0.41199,
        y: -0.55562,
        z: 0.77094
    },
    compass: {
        x: -106.50102,
        y: -80.6425,
        z: -36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 212.94773,
        y: 147.81605,
        z: 16.67983
    },
    accel: {
        x: -0.42173,
        y: -0.62757,
        z: 0.8125
    },
    compass: {
        x: -99.09226,
        y: -80.6425,
        z: -36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 215.33704,
        y: 159.05269,
        z: 28.16838
    },
    accel: {
        x: -0.34625,
        y: -0.91319,
        z: 0.745
    },
    compass: {
        x: -100.94445,
        y: -82.43456,
        z: -40.28229
    },
    dt: 13
}, {
    gyro: {
        x: 227.55841,
        y: 160.40383,
        z: 35.44319
    },
    accel: {
        x: -0.43463,
        y: -0.63075,
        z: 0.66837
    },
    compass: {
        x: -97.24006,
        y: -84.22661,
        z: -40.28229
    },
    dt: 13
}, {
    gyro: {
        x: 236.18437,
        y: 162.12902,
        z: 36.17602
    },
    accel: {
        x: -0.40809,
        y: -0.55806,
        z: 0.67173
    },
    compass: {
        x: -93.53568,
        y: -87.81072,
        z: -41.99643
    },
    dt: 15
}, {
    gyro: {
        x: 248.35994,
        y: 164.3275,
        z: 31.8096
    },
    accel: {
        x: -0.40834,
        y: -0.47485,
        z: 0.64843
    },
    compass: {
        x: -87.97911,
        y: -87.81072,
        z: -47.13885
    },
    dt: 13
}, {
    gyro: {
        x: 248.75689,
        y: 162.96108,
        z: 24.91647
    },
    accel: {
        x: -0.43877,
        y: -0.36887,
        z: 0.62801
    },
    compass: {
        x: -86.12691,
        y: -93.18689,
        z: -43.71057
    },
    dt: 12
}, {
    gyro: {
        x: 250.45154,
        y: 158.61757,
        z: 15.20655
    },
    accel: {
        x: -0.48479,
        y: -0.31723,
        z: 0.64074
    },
    compass: {
        x: -86.12691,
        y: -93.18689,
        z: -43.71057
    },
    dt: 13
}, {
    gyro: {
        x: 242.06986,
        y: 158.75498,
        z: 6.27525
    },
    accel: {
        x: -0.46215,
        y: -0.38429,
        z: 0.63258
    },
    compass: {
        x: -78.71815,
        y: -93.18689,
        z: -45.42471
    },
    dt: 13
}, {
    gyro: {
        x: 234.52024,
        y: 155.36566,
        z: -0.59498
    },
    accel: {
        x: -0.56612,
        y: -0.27146,
        z: 0.61216
    },
    compass: {
        x: -78.71815,
        y: -94.97895,
        z: -45.42471
    },
    dt: 12
}, {
    gyro: {
        x: 228.26834,
        y: 154.27406,
        z: -13.2591
    },
    accel: {
        x: -0.54761,
        y: -0.0879,
        z: 0.68783
    },
    compass: {
        x: -75.01376,
        y: -100.35511,
        z: -47.13885
    },
    dt: 12
}, {
    gyro: {
        x: 230.72635,
        y: 153.34276,
        z: -29.57971
    },
    accel: {
        x: -0.60167,
        y: -0.03039,
        z: 0.68446
    },
    compass: {
        x: -69.45719,
        y: -102.14717,
        z: -48.85299
    },
    dt: 13
}, {
    gyro: {
        x: 239.50498,
        y: 142.04505,
        z: -44.64078
    },
    accel: {
        x: -0.70272,
        y: -0.00442,
        z: 0.70608
    },
    compass: {
        x: -69.45719,
        y: -102.14717,
        z: -48.85299
    },
    dt: 12
}, {
    gyro: {
        x: 241.89429,
        y: 131.62521,
        z: -54.42704
    },
    accel: {
        x: -0.76091,
        y: -0.21101,
        z: 0.82668
    },
    compass: {
        x: -71.30938,
        y: -103.93922,
        z: -50.56713
    },
    dt: 13
}, {
    gyro: {
        x: 243.77215,
        y: 130.67101,
        z: -56.63315
    },
    accel: {
        x: -0.76554,
        y: -0.30426,
        z: 0.86511
    },
    compass: {
        x: -65.75281,
        y: -103.93922,
        z: -48.85299
    },
    dt: 12
}, {
    gyro: {
        x: 234.29887,
        y: 137.13666,
        z: -62.12933
    },
    accel: {
        x: -0.71148,
        y: -0.25531,
        z: 0.77239
    },
    compass: {
        x: -67.605,
        y: -105.73128,
        z: -50.56713
    },
    dt: 13
}, {
    gyro: {
        x: 250.45154,
        y: 125.15192,
        z: -71.12933
    },
    accel: {
        x: -0.78234,
        y: -0.16598,
        z: 0.54081
    },
    compass: {
        x: -67.605,
        y: -105.73128,
        z: -50.56713
    },
    dt: 13
}, {
    gyro: {
        x: 250.45154,
        y: 96.54887,
        z: -73.99192
    },
    accel: {
        x: -0.75312,
        y: -0.02427,
        z: 0.43319
    },
    compass: {
        x: -65.75281,
        y: -102.14717,
        z: -50.56713
    },
    dt: 12
}, {
    gyro: {
        x: 250.45154,
        y: 66.3275,
        z: -74.27437
    },
    accel: {
        x: -0.76773,
        y: -0.19119,
        z: 0.38731
    },
    compass: {
        x: -60.19623,
        y: -105.73128,
        z: -53.99541
    },
    dt: 12
}, {
    gyro: {
        x: 250.45154,
        y: 34.91528,
        z: -73.63315
    },
    accel: {
        x: -0.70539,
        y: -0.25898,
        z: 0.2917
    },
    compass: {
        x: -62.04842,
        y: -102.14717,
        z: -53.99541
    },
    dt: 13
}, {
    gyro: {
        x: 218.06223,
        y: 13.33513,
        z: -77.84689
    },
    accel: {
        x: -0.63332,
        y: -0.17528,
        z: 0.38539
    },
    compass: {
        x: -62.04842,
        y: -102.14717,
        z: -53.99541
    },
    dt: 13
}, {
    gyro: {
        x: 181.2149,
        y: 15.08322,
        z: -71.94612
    },
    accel: {
        x: -0.67958,
        y: -0.31846,
        z: 0.42574
    },
    compass: {
        x: -60.19623,
        y: -103.93922,
        z: -64.28026
    },
    dt: 12
}, {
    gyro: {
        x: 143.90956,
        y: 20.03742,
        z: -49.32017
    },
    accel: {
        x: -0.70247,
        y: -0.39849,
        z: 0.38034
    },
    compass: {
        x: -58.34404,
        y: -102.14717,
        z: -65.9944
    },
    dt: 13
}, {
    gyro: {
        x: 117.08513,
        y: 22.99925,
        z: -47.06063
    },
    accel: {
        x: -0.72828,
        y: -0.25923,
        z: 0.40292
    },
    compass: {
        x: -56.49185,
        y: -100.35511,
        z: -62.56612
    },
    dt: 12
}, {
    gyro: {
        x: 89.24544,
        y: 38.93055,
        z: -62.99956
    },
    accel: {
        x: -0.74386,
        y: -0.27807,
        z: 0.49205
    },
    compass: {
        x: -56.49185,
        y: -100.35511,
        z: -62.56612
    },
    dt: 12
}, {
    gyro: {
        x: 91.77979,
        y: 57.20536,
        z: -48.62551
    },
    accel: {
        x: -0.67642,
        y: -0.43398,
        z: 0.55715
    },
    compass: {
        x: -56.49185,
        y: -100.35511,
        z: -62.56612
    },
    dt: 13
}, {
    gyro: {
        x: 96.94773,
        y: 71.80841,
        z: -21.2133
    },
    accel: {
        x: -0.82202,
        y: -0.40926,
        z: 0.40797
    },
    compass: {
        x: -60.19623,
        y: -103.93922,
        z: -64.28026
    },
    dt: 13
}, {
    gyro: {
        x: 93.90956,
        y: 74.67864,
        z: -12.61788
    },
    accel: {
        x: -0.74679,
        y: -0.38405,
        z: 0.32317
    },
    compass: {
        x: -60.19623,
        y: -102.14717,
        z: -69.42268
    },
    dt: 12
}, {
    gyro: {
        x: 96.2607,
        y: 88.73208,
        z: -15.03773
    },
    accel: {
        x: -0.68884,
        y: -0.29447,
        z: 0.51126
    },
    compass: {
        x: -60.19623,
        y: -102.14717,
        z: -69.42268
    },
    dt: 13
}, {
    gyro: {
        x: 93.58131,
        y: 116.50307,
        z: -18.58734
    },
    accel: {
        x: -0.68494,
        y: -0.37695,
        z: 0.45433
    },
    compass: {
        x: -60.19623,
        y: -102.14717,
        z: -69.42268
    },
    dt: 13
}, {
    gyro: {
        x: 77.36757,
        y: 119.31986,
        z: -11.17513
    },
    accel: {
        x: -0.69833,
        y: -0.36961,
        z: 0.19369
    },
    compass: {
        x: -54.63965,
        y: -105.73128,
        z: -62.56612
    },
    dt: 13
}, {
    gyro: {
        x: 74.15383,
        y: 99.27406,
        z: -12.07589
    },
    accel: {
        x: -0.68664,
        y: -0.20171,
        z: 0.09472
    },
    compass: {
        x: -52.78746,
        y: -107.52334,
        z: -62.56612
    },
    dt: 12
}, {
    gyro: {
        x: 73.89429,
        y: 79.77788,
        z: -15.44231
    },
    accel: {
        x: -0.79208,
        y: -0.10602,
        z: 0.19993
    },
    compass: {
        x: -52.78746,
        y: -107.52334,
        z: -62.56612
    },
    dt: 12
}, {
    gyro: {
        x: 96.81032,
        y: 71.12902,
        z: -10.19803
    },
    accel: {
        x: -0.88387,
        y: -0.19363,
        z: 0.3102
    },
    compass: {
        x: -52.78746,
        y: -107.52334,
        z: -62.56612
    },
    dt: 13
}, {
    gyro: {
        x: 80.31414,
        y: 65.09085,
        z: 6.06151
    },
    accel: {
        x: -0.97201,
        y: -0.15521,
        z: 0.35512
    },
    compass: {
        x: -49.08308,
        y: -111.10745,
        z: -59.13783
    },
    dt: 12
}, {
    gyro: {
        x: 60.53551,
        y: 81.59467,
        z: 8.09968
    },
    accel: {
        x: -0.9708,
        y: -0.02451,
        z: 0.52279
    },
    compass: {
        x: -50.93527,
        y: -111.10745,
        z: -57.42369
    },
    dt: 12
}, {
    gyro: {
        x: 42.69582,
        y: 91.80078,
        z: 5.46609
    },
    accel: {
        x: -0.99344,
        y: -0.06245,
        z: 0.28858
    },
    compass: {
        x: -47.23089,
        y: -112.8995,
        z: -57.42369
    },
    dt: 13
}, {
    gyro: {
        x: 17.76452,
        y: 63.4191,
        z: -5.09879
    },
    accel: {
        x: -0.99076,
        y: 0.15637,
        z: 0.10144
    },
    compass: {
        x: -45.3787,
        y: -112.8995,
        z: -52.28127
    },
    dt: 21
}, {
    gyro: {
        x: 16.80269,
        y: 45.28169,
        z: -17.92322
    },
    accel: {
        x: -1.0453,
        y: 0.15662,
        z: 0.29122
    },
    compass: {
        x: -45.3787,
        y: -112.8995,
        z: -52.28127
    },
    dt: 13
}, {
    gyro: {
        x: 14.35231,
        y: 43.81605,
        z: -22.05299
    },
    accel: {
        x: -1.0699,
        y: 0.19043,
        z: 0.50862
    },
    compass: {
        x: -45.3787,
        y: -118.27567,
        z: -48.85299
    },
    dt: 12
}, {
    gyro: {
        x: 35.79505,
        y: 50.60994,
        z: -25.03009
    },
    accel: {
        x: -1.1108,
        y: 0.30938,
        z: 0.53457
    },
    compass: {
        x: -45.3787,
        y: -120.06773,
        z: -47.13885
    },
    dt: 13
}, {
    gyro: {
        x: 23.50498,
        y: 58.22826,
        z: -34.89269
    },
    accel: {
        x: -1.10934,
        y: 0.3483,
        z: 0.58693
    },
    compass: {
        x: -47.23089,
        y: -114.69156,
        z: -47.13885
    },
    dt: 12
}, {
    gyro: {
        x: 28.35994,
        y: 75.56414,
        z: -47.20566
    },
    accel: {
        x: -1.1052,
        y: 0.31619,
        z: 0.64387
    },
    compass: {
        x: -47.23089,
        y: -114.69156,
        z: -47.13885
    },
    dt: 12
}, {
    gyro: {
        x: 22.39811,
        y: 96.4191,
        z: -56.03009
    },
    accel: {
        x: -1.11567,
        y: 0.25538,
        z: 0.53553
    },
    compass: {
        x: -47.23089,
        y: -114.69156,
        z: -47.13885
    },
    dt: 13
}, {
    gyro: {
        x: 9.45154,
        y: 113.13666,
        z: -57.3278
    },
    accel: {
        x: -1.07379,
        y: 0.06515,
        z: 0.38803
    },
    compass: {
        x: -52.78746,
        y: -120.06773,
        z: -41.99643
    },
    dt: 12
}, {
    gyro: {
        x: 3.192,
        y: 118.70154,
        z: -47.98429
    },
    accel: {
        x: -1.09035,
        y: 0.03766,
        z: 0.16102
    },
    compass: {
        x: -52.78746,
        y: -118.27567,
        z: -40.28229
    },
    dt: 13
}, {
    gyro: {
        x: -5.70876,
        y: 115.48017,
        z: -40.83925
    },
    accel: {
        x: -1.07501,
        y: 0.18751,
        z: 0.01881
    },
    compass: {
        x: -52.78746,
        y: -120.06773,
        z: -35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -5.43395,
        y: 110.24353,
        z: -45.40414
    },
    accel: {
        x: -1.03678,
        y: 0.323,
        z: -0.01798
    },
    compass: {
        x: -52.78746,
        y: -120.06773,
        z: -35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 6.2836,
        y: 106.38093,
        z: -60.25147
    },
    accel: {
        x: -1.00294,
        y: 0.37944,
        z: -0.02303
    },
    compass: {
        x: -52.78746,
        y: -120.06773,
        z: -26.56917
    },
    dt: 12
}, {
    gyro: {
        x: 10.60421,
        y: 103.70918,
        z: -73.03009
    },
    accel: {
        x: -1.00927,
        y: 0.21622,
        z: -0.03939
    },
    compass: {
        x: -54.63965,
        y: -125.44389,
        z: -26.56917
    },
    dt: 13
}, {
    gyro: {
        x: 14.68055,
        y: 102.78551,
        z: -69.44231
    },
    accel: {
        x: -1.03678,
        y: 0.09726,
        z: -0.02688
    },
    compass: {
        x: -60.19623,
        y: -116.48361,
        z: -21.42675
    },
    dt: 13
}, {
    gyro: {
        x: 13.98589,
        y: 105.22063,
        z: -73.44994
    },
    accel: {
        x: -1.00221,
        y: 0.2094,
        z: -0.00474
    },
    compass: {
        x: -60.19623,
        y: -116.48361,
        z: -21.42675
    },
    dt: 12
}, {
    gyro: {
        x: 10.77979,
        y: 110.3504,
        z: -81.54154
    },
    accel: {
        x: -1.00415,
        y: 0.10164,
        z: -0.0122
    },
    compass: {
        x: -62.04842,
        y: -121.85978,
        z: -17.99847
    },
    dt: 13
}, {
    gyro: {
        x: 3.48971,
        y: 117.50307,
        z: -82.98429
    },
    accel: {
        x: -0.98784,
        y: -0.02549,
        z: -0.0329
    },
    compass: {
        x: -65.75281,
        y: -120.06773,
        z: -12.85605
    },
    dt: 12
}, {
    gyro: {
        x: -5.43395,
        y: 123.87711,
        z: -78.36597
    },
    accel: {
        x: -0.90627,
        y: -0.07567,
        z: -0.08896
    },
    compass: {
        x: -71.30938,
        y: -118.27567,
        z: -14.57019
    },
    dt: 13
}, {
    gyro: {
        x: -16.52556,
        y: 124.96108,
        z: -70.36597
    },
    accel: {
        x: -0.86731,
        y: -0.17234,
        z: -0.17727
    },
    compass: {
        x: -71.30938,
        y: -118.27567,
        z: -14.57019
    },
    dt: 12
}, {
    gyro: {
        x: -21.28128,
        y: 122.12139,
        z: -55.08353
    },
    accel: {
        x: -0.84467,
        y: -0.30402,
        z: -0.26005
    },
    compass: {
        x: -73.16157,
        y: -118.27567,
        z: -5.99949
    },
    dt: 13
}, {
    gyro: {
        x: -0.46449,
        y: 112.25879,
        z: -50.03009
    },
    accel: {
        x: -0.72463,
        y: -0.14175,
        z: -0.30553
    },
    compass: {
        x: -75.01376,
        y: -111.10745,
        z: -0.85707
    },
    dt: 12
}, {
    gyro: {
        x: -7.15151,
        y: 99.5794,
        z: -57.4194
    },
    accel: {
        x: -0.79597,
        y: -0.22717,
        z: -0.39697
    },
    compass: {
        x: -75.01376,
        y: -118.27567,
        z: 2.57121
    },
    dt: 13
}, {
    gyro: {
        x: -30.01411,
        y: 93.52597,
        z: -57.95376
    },
    accel: {
        x: -0.69736,
        y: -0.36324,
        z: -0.37892
    },
    compass: {
        x: -75.01376,
        y: -118.27567,
        z: 2.57121
    },
    dt: 13
}, {
    gyro: {
        x: -35.05991,
        y: 84.47253,
        z: -57.74002
    },
    accel: {
        x: -0.7256,
        y: -0.27416,
        z: -0.43354
    },
    compass: {
        x: -78.71815,
        y: -112.8995,
        z: 11.14191
    },
    dt: 13
}, {
    gyro: {
        x: -40.28892,
        y: 69.34276,
        z: -60.90032
    },
    accel: {
        x: -0.75531,
        y: -0.16402,
        z: -0.42993
    },
    compass: {
        x: -76.86596,
        y: -112.8995,
        z: 9.42777
    },
    dt: 12
}, {
    gyro: {
        x: -43.22021,
        y: 61.48017,
        z: -66.03773
    },
    accel: {
        x: -0.74825,
        y: -0.32727,
        z: -0.36689
    },
    compass: {
        x: -78.71815,
        y: -112.8995,
        z: 16.28433
    },
    dt: 13
}, {
    gyro: {
        x: -62.6935,
        y: 62.5565,
        z: -62.91559
    },
    accel: {
        x: -0.74995,
        y: -0.53432,
        z: -0.4102
    },
    compass: {
        x: -78.71815,
        y: -112.8995,
        z: 16.28433
    },
    dt: 13
}, {
    gyro: {
        x: -71.86144,
        y: 57.29696,
        z: -49.49574
    },
    accel: {
        x: -0.73364,
        y: -0.52771,
        z: -0.42512
    },
    compass: {
        x: -86.12691,
        y: -111.10745,
        z: 16.28433
    },
    dt: 13
}, {
    gyro: {
        x: -70.83853,
        y: 50.27406,
        z: -38.96139
    },
    accel: {
        x: -0.6864,
        y: -0.44842,
        z: -0.46506
    },
    compass: {
        x: -84.27472,
        y: -109.31539,
        z: 17.99847
    },
    dt: 12
}, {
    gyro: {
        x: -73.83853,
        y: 36.78551,
        z: -44.24383
    },
    accel: {
        x: -0.70515,
        y: -0.48023,
        z: -0.59236
    },
    compass: {
        x: -87.97911,
        y: -109.31539,
        z: 21.42675
    },
    dt: 20
}, {
    gyro: {
        x: -74.97594,
        y: 17.5107,
        z: -37.74002
    },
    accel: {
        x: -0.70807,
        y: -0.68191,
        z: -0.62243
    },
    compass: {
        x: -87.97911,
        y: -107.52334,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: -70.68586,
        y: 4.49544,
        z: -21.85452
    },
    accel: {
        x: -0.692,
        y: -0.66306,
        z: -0.52739
    },
    compass: {
        x: -87.97911,
        y: -107.52334,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: -72.20495,
        y: 2.29696,
        z: -10.12169
    },
    accel: {
        x: -0.70685,
        y: -0.53946,
        z: -0.49755
    },
    compass: {
        x: -87.97911,
        y: -107.52334,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: -86.8538,
        y: 0.93818,
        z: -7.58734
    },
    accel: {
        x: -0.69225,
        y: -0.54191,
        z: -0.64265
    },
    compass: {
        x: -82.42253,
        y: -105.73128,
        z: 31.71159
    },
    dt: 12
}, {
    gyro: {
        x: -110.09044,
        y: -8.78701,
        z: -6.28963
    },
    accel: {
        x: -0.62553,
        y: -0.50055,
        z: -0.78101
    },
    compass: {
        x: -87.97911,
        y: -105.73128,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: -115.77747,
        y: -29.97021,
        z: -8.78582
    },
    accel: {
        x: -0.66984,
        y: -0.30353,
        z: -0.90301
    },
    compass: {
        x: -87.97911,
        y: -105.73128,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -111.60189,
        y: -54.41296,
        z: -11.53391
    },
    accel: {
        x: -0.69273,
        y: -0.29643,
        z: -0.95426
    },
    compass: {
        x: -87.97911,
        y: -105.73128,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -123.67823,
        y: -77.5122,
        z: -9.63315
    },
    accel: {
        x: -0.69298,
        y: -0.41929,
        z: -0.98145
    },
    compass: {
        x: -82.42253,
        y: -105.73128,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -149.18968,
        y: -103.5809,
        z: -9.60261
    },
    accel: {
        x: -0.6903,
        y: -0.45086,
        z: -0.96437
    },
    compass: {
        x: -84.27472,
        y: -103.93922,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -188.41869,
        y: -127.92441,
        z: -14.03773
    },
    accel: {
        x: -0.68616,
        y: -0.5353,
        z: -0.89699
    },
    compass: {
        x: -82.42253,
        y: -103.93922,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -227.81563,
        y: -148.76411,
        z: -19.85452
    },
    accel: {
        x: -0.6601,
        y: -0.26412,
        z: -0.78678
    },
    compass: {
        x: -76.86596,
        y: -107.52334,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: -223.26602,
        y: -180.38243,
        z: -33.45757
    },
    accel: {
        x: -0.78794,
        y: -0.04972,
        z: -0.64505
    },
    compass: {
        x: -76.86596,
        y: -107.52334,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: -208.8538,
        y: -190.85571,
        z: -35.46521
    },
    accel: {
        x: -0.7424,
        y: -0.21077,
        z: -0.42007
    },
    compass: {
        x: -76.86596,
        y: -111.10745,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -203.6935,
        y: -189.00838,
        z: -35.47284
    },
    accel: {
        x: -0.78915,
        y: -0.24993,
        z: -0.40828
    },
    compass: {
        x: -78.71815,
        y: -107.52334,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -201.30418,
        y: -190.26029,
        z: -38.86979
    },
    accel: {
        x: -0.75993,
        y: -0.1229,
        z: -0.3948
    },
    compass: {
        x: -78.71815,
        y: -107.52334,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -197.23548,
        y: -196.07708,
        z: -52.76292
    },
    accel: {
        x: -0.75847,
        y: -0.01791,
        z: -0.37603
    },
    compass: {
        x: -78.71815,
        y: -107.52334,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -213.26602,
        y: -194.78701,
        z: -68.27437
    },
    accel: {
        x: -0.68859,
        y: -0.11899,
        z: -0.33416
    },
    compass: {
        x: -78.71815,
        y: -109.31539,
        z: 28.28331
    },
    dt: 13
}, {
    gyro: {
        x: -242.31945,
        y: -193.81754,
        z: -76.22093
    },
    accel: {
        x: -0.6976,
        y: -0.07102,
        z: -0.42873
    },
    compass: {
        x: -75.01376,
        y: -109.31539,
        z: 28.28331
    },
    dt: 13
}, {
    gyro: {
        x: -249.81563,
        y: -214.17632,
        z: -85.09879
    },
    accel: {
        x: -0.72439,
        y: 0.07026,
        z: -0.48816
    },
    compass: {
        x: -75.01376,
        y: -109.31539,
        z: 28.28331
    },
    dt: 12
}, {
    gyro: {
        x: -249.81563,
        y: -236.61907,
        z: -103.57208
    },
    accel: {
        x: -0.65207,
        y: 0.01893,
        z: -0.40804
    },
    compass: {
        x: -75.01376,
        y: -109.31539,
        z: 28.28331
    },
    dt: 12
}, {
    gyro: {
        x: -249.81563,
        y: -239.88624,
        z: -133.00719
    },
    accel: {
        x: -0.54713,
        y: -0.13661,
        z: -0.29999
    },
    compass: {
        x: -80.57034,
        y: -105.73128,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: -249.81563,
        y: -240.93205,
        z: -151.17513
    },
    accel: {
        x: -0.55662,
        y: -0.45674,
        z: -0.32983
    },
    compass: {
        x: -80.57034,
        y: -109.31539,
        z: 24.85503
    },
    dt: 13
}, {
    gyro: {
        x: -249.81563,
        y: -251.32899,
        z: -156.93849
    },
    accel: {
        x: -0.52497,
        y: -0.61215,
        z: -0.36352
    },
    compass: {
        x: -80.57034,
        y: -109.31539,
        z: 24.85503
    },
    dt: 13
}, {
    gyro: {
        x: -249.81563,
        y: -251.32899,
        z: -162.95376
    },
    accel: {
        x: -0.50719,
        y: -0.5752,
        z: -0.24417
    },
    compass: {
        x: -80.57034,
        y: -109.31539,
        z: 24.85503
    },
    dt: 13
}, {
    gyro: {
        x: -249.81563,
        y: -251.32899,
        z: -167.64841
    },
    accel: {
        x: -0.50841,
        y: -0.5796,
        z: 0.07646
    },
    compass: {
        x: -87.97911,
        y: -98.56306,
        z: 24.85503
    },
    dt: 13
}, {
    gyro: {
        x: -249.81563,
        y: -251.32899,
        z: -175.3507
    },
    accel: {
        x: -0.56904,
        y: -0.40143,
        z: 0.38731
    },
    compass: {
        x: -87.97911,
        y: -102.14717,
        z: 23.14089
    },
    dt: 12
}, {
    gyro: {
        x: -249.81563,
        y: -251.32899,
        z: -181.65605
    },
    accel: {
        x: -0.71197,
        y: -0.43569,
        z: 0.69239
    },
    compass: {
        x: -87.97911,
        y: -102.14717,
        z: 23.14089
    },
    dt: 13
}, {
    gyro: {
        x: -216.28128,
        y: -215.38243,
        z: -167.70185
    },
    accel: {
        x: -0.74119,
        y: -0.70491,
        z: 0.84181
    },
    compass: {
        x: -95.38787,
        y: -96.771,
        z: 21.42675
    },
    dt: 13
}, {
    gyro: {
        x: -189.36525,
        y: -157.64197,
        z: -145.68658
    },
    accel: {
        x: -0.72146,
        y: -0.81236,
        z: 0.69599
    },
    compass: {
        x: -97.24006,
        y: -93.18689,
        z: 21.42675
    },
    dt: 13
}, {
    gyro: {
        x: -180.00647,
        y: -120.08472,
        z: -134.64078
    },
    accel: {
        x: -0.63551,
        y: -0.84246,
        z: 0.3532
    },
    compass: {
        x: -100.94445,
        y: -91.39484,
        z: 23.14089
    },
    dt: 13
}, {
    gyro: {
        x: -181.42632,
        y: -110.34426,
        z: -131.73238
    },
    accel: {
        x: -0.5666,
        y: -0.80477,
        z: 0.08054
    },
    compass: {
        x: -100.94445,
        y: -91.39484,
        z: 23.14089
    },
    dt: 12
}, {
    gyro: {
        x: -179.93777,
        y: -121.94731,
        z: -131.08353
    },
    accel: {
        x: -0.53763,
        y: -0.75068,
        z: -0.07838
    },
    compass: {
        x: -100.94445,
        y: -91.39484,
        z: 23.14089
    },
    dt: 12
}, {
    gyro: {
        x: -181.8996,
        y: -140.68014,
        z: -137.89269
    },
    accel: {
        x: -0.48698,
        y: -0.55341,
        z: -0.03025
    },
    compass: {
        x: -104.64883,
        y: -82.43456,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: -179.04464,
        y: -154.29082,
        z: -153.74765
    },
    accel: {
        x: -0.50427,
        y: -0.54558,
        z: 0.12378
    },
    compass: {
        x: -106.50102,
        y: -84.22661,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: -163.93777,
        y: -142.79464,
        z: -145.84689
    },
    accel: {
        x: -0.53739,
        y: -0.89802,
        z: 0.41277
    },
    compass: {
        x: -108.35321,
        y: -77.05839,
        z: 29.99745
    },
    dt: 19
}, {
    gyro: {
        x: -160.39579,
        y: -115.26792,
        z: -138.70185
    },
    accel: {
        x: -0.49916,
        y: -0.91442,
        z: 0.43223
    },
    compass: {
        x: -110.20541,
        y: -75.26634,
        z: 33.42573
    },
    dt: 48
}, {
    gyro: {
        x: -144.33472,
        y: -90.58853,
        z: -91.89269
    },
    accel: {
        x: -0.42806,
        y: -0.96728,
        z: 0.12859
    },
    compass: {
        x: -108.35321,
        y: -69.89017,
        z: 40.28229
    },
    dt: 13
}, {
    gyro: {
        x: -148.9454,
        y: -87.81754,
        z: -96.84689
    },
    accel: {
        x: -0.41467,
        y: -0.84638,
        z: 0.10841
    },
    compass: {
        x: -112.0576,
        y: -64.514,
        z: 47.13885
    },
    dt: 40
}, {
    gyro: {
        x: -101.54846,
        y: -72.72594,
        z: -99.48047
    },
    accel: {
        x: -0.32799,
        y: -0.90952,
        z: 0.14228
    },
    compass: {
        x: -112.0576,
        y: -64.514,
        z: 50.56713
    },
    dt: 12
}, {
    gyro: {
        x: -82.28892,
        y: -61.68777,
        z: -92.29727
    },
    accel: {
        x: -0.33213,
        y: -1.0456,
        z: 0.03875
    },
    compass: {
        x: -112.0576,
        y: -64.514,
        z: 50.56713
    },
    dt: 12
}, {
    gyro: {
        x: -63.7851,
        y: -59.54273,
        z: -76.07589
    },
    accel: {
        x: -0.30607,
        y: -1.0921,
        z: -0.07164
    },
    compass: {
        x: -110.20541,
        y: -60.92989,
        z: 52.28127
    },
    dt: 14
}, {
    gyro: {
        x: -53.808,
        y: -60.54273,
        z: -58.18276
    },
    accel: {
        x: -0.28075,
        y: -1.09333,
        z: -0.10027
    },
    compass: {
        x: -110.20541,
        y: -59.13783,
        z: 57.42369
    },
    dt: 13
}, {
    gyro: {
        x: -41.51029,
        y: -58.48166,
        z: -39.1904
    },
    accel: {
        x: -0.2659,
        y: -1.06689,
        z: -0.1034
    },
    compass: {
        x: -108.35321,
        y: -53.76167,
        z: 53.99541
    },
    dt: 12
}, {
    gyro: {
        x: -32.51029,
        y: -51.79464,
        z: -28.19803
    },
    accel: {
        x: -0.14683,
        y: -0.90414,
        z: -0.09522
    },
    compass: {
        x: -108.35321,
        y: -53.76167,
        z: 53.99541
    },
    dt: 13
}, {
    gyro: {
        x: -29.19731,
        y: -42.95495,
        z: -31.282
    },
    accel: {
        x: -0.20088,
        y: -0.92812,
        z: -0.18906
    },
    compass: {
        x: -110.20541,
        y: -55.55372,
        z: 53.99541
    },
    dt: 13
}, {
    gyro: {
        x: -38.02937,
        y: -38.13815,
        z: -28.36597
    },
    accel: {
        x: -0.17118,
        y: -1.01672,
        z: -0.27617
    },
    compass: {
        x: -108.35321,
        y: -59.13783,
        z: 52.28127
    },
    dt: 12
}, {
    gyro: {
        x: -42.75456,
        y: -43.03128,
        z: -34.18276
    },
    accel: {
        x: -0.12126,
        y: -0.90928,
        z: -0.42271
    },
    compass: {
        x: -110.20541,
        y: -59.13783,
        z: 52.28127
    },
    dt: 12
}, {
    gyro: {
        x: -78.13624,
        y: -63.74121,
        z: -51.64841
    },
    accel: {
        x: -0.07938,
        y: -0.81847,
        z: -0.45448
    },
    compass: {
        x: -110.20541,
        y: -59.13783,
        z: 52.28127
    },
    dt: 13
}, {
    gyro: {
        x: -90.09808,
        y: -88.45113,
        z: -60.98429
    },
    accel: {
        x: -0.16534,
        y: -0.89263,
        z: -0.40394
    },
    compass: {
        x: -110.20541,
        y: -59.13783,
        z: 52.28127
    },
    dt: 12
}, {
    gyro: {
        x: -67.67823,
        y: -108.68014,
        z: -59.4194
    },
    accel: {
        x: -0.19796,
        y: -0.96165,
        z: -0.26294
    },
    compass: {
        x: -110.20541,
        y: -59.13783,
        z: 57.42369
    },
    dt: 12
}, {
    gyro: {
        x: -41.93777,
        y: -112.43586,
        z: -54.2362
    },
    accel: {
        x: -0.14464,
        y: -0.99078,
        z: -0.12385
    },
    compass: {
        x: -110.20541,
        y: -53.76167,
        z: 57.42369
    },
    dt: 12
}, {
    gyro: {
        x: -25.82327,
        y: -105.12289,
        z: -45.11406
    },
    accel: {
        x: -0.15438,
        y: -1.01843,
        z: -0.11591
    },
    compass: {
        x: -106.50102,
        y: -55.55372,
        z: 57.42369
    },
    dt: 12
}, {
    gyro: {
        x: -21.03701,
        y: -100.07708,
        z: -32.33544
    },
    accel: {
        x: -0.15681,
        y: -0.95969,
        z: -0.17415
    },
    compass: {
        x: -110.20541,
        y: -55.55372,
        z: 59.13783
    },
    dt: 13
}, {
    gyro: {
        x: -27.25838,
        y: -112.91678,
        z: -37.19803
    },
    accel: {
        x: -0.15633,
        y: -0.80771,
        z: -0.19195
    },
    compass: {
        x: -110.20541,
        y: -57.34578,
        z: 55.70955
    },
    dt: 20
}, {
    gyro: {
        x: -55.63243,
        y: -121.4664,
        z: -50.91559
    },
    accel: {
        x: -0.16144,
        y: -0.90169,
        z: 0.00224
    },
    compass: {
        x: -110.20541,
        y: -57.34578,
        z: 55.70955
    },
    dt: 13
}, {
    gyro: {
        x: -54.86907,
        y: -108.90151,
        z: -58.74765
    },
    accel: {
        x: -0.16947,
        y: -0.85078,
        z: 0.13027
    },
    compass: {
        x: -108.35321,
        y: -62.72195,
        z: 53.99541
    },
    dt: 12
}, {
    gyro: {
        x: -36.63243,
        y: -90.73357,
        z: -65.76292
    },
    accel: {
        x: -0.18311,
        y: -0.81945,
        z: 0.17087
    },
    compass: {
        x: -112.0576,
        y: -57.34578,
        z: 53.99541
    },
    dt: 13
}, {
    gyro: {
        x: -11.26602,
        y: -74.92441,
        z: -71.49574
    },
    accel: {
        x: -0.14269,
        y: -0.7639,
        z: 0.15189
    },
    compass: {
        x: -112.0576,
        y: -57.34578,
        z: 53.99541
    },
    dt: 13
}, {
    gyro: {
        x: 6.51261,
        y: -66.08472,
        z: -78.77818
    },
    accel: {
        x: -0.13539,
        y: -0.83634,
        z: 0.05316
    },
    compass: {
        x: -112.0576,
        y: -57.34578,
        z: 53.99541
    },
    dt: 12
}, {
    gyro: {
        x: 12.20727,
        y: -64.4664,
        z: -80.06063
    },
    accel: {
        x: -0.07768,
        y: -0.95969,
        z: -0.07621
    },
    compass: {
        x: -113.90979,
        y: -53.76167,
        z: 48.85299
    },
    dt: 12
}, {
    gyro: {
        x: 11.84849,
        y: -78.25266,
        z: -79.13696
    },
    accel: {
        x: -0.0901,
        y: -0.92983,
        z: -0.16091
    },
    compass: {
        x: -115.76198,
        y: -57.34578,
        z: 52.28127
    },
    dt: 12
}, {
    gyro: {
        x: 0.93246,
        y: -108.18395,
        z: -80.09116
    },
    accel: {
        x: -0.17824,
        y: -1.10312,
        z: 0.02193
    },
    compass: {
        x: -115.76198,
        y: -55.55372,
        z: 50.56713
    },
    dt: 13
}, {
    gyro: {
        x: 1.72635,
        y: -106.88624,
        z: -62.13696
    },
    accel: {
        x: -0.15852,
        y: -1.2343,
        z: 0.35128
    },
    compass: {
        x: -115.76198,
        y: -55.55372,
        z: 50.56713
    },
    dt: 13
}, {
    gyro: {
        x: 10.97826,
        y: -76.71067,
        z: -44.87742
    },
    accel: {
        x: -0.11859,
        y: -1.08011,
        z: 0.37314
    },
    compass: {
        x: -119.46636,
        y: -53.76167,
        z: 45.42471
    },
    dt: 13
}, {
    gyro: {
        x: 14.25307,
        y: -50.32899,
        z: -45.72475
    },
    accel: {
        x: -0.11542,
        y: -0.97683,
        z: 0.30011
    },
    compass: {
        x: -117.61417,
        y: -50.17756,
        z: 45.42471
    },
    dt: 12
}, {
    gyro: {
        x: 9.16147,
        y: -39.81754,
        z: -55.3736
    },
    accel: {
        x: -0.03604,
        y: -0.94232,
        z: 0.10312
    },
    compass: {
        x: -119.46636,
        y: -51.96961,
        z: 45.42471
    },
    dt: 13
}, {
    gyro: {
        x: 12.24544,
        y: -48.16105,
        z: -66.2133
    },
    accel: {
        x: 0.00073,
        y: -0.99592,
        z: -0.04445
    },
    compass: {
        x: -119.46636,
        y: -51.96961,
        z: 45.42471
    },
    dt: 12
}, {
    gyro: {
        x: 11.27597,
        y: -68.61144,
        z: -67.93849
    },
    accel: {
        x: -0.0789,
        y: -1.18364,
        z: -0.09353
    },
    compass: {
        x: -119.46636,
        y: -51.96961,
        z: 45.42471
    },
    dt: 12
}, {
    gyro: {
        x: 11.96299,
        y: -86.50456,
        z: -56.36597
    },
    accel: {
        x: -0.10179,
        y: -1.30161,
        z: 0.08607
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 41.99643
    },
    dt: 12
}, {
    gyro: {
        x: 17.39047,
        y: -83.68777,
        z: -35.2591
    },
    accel: {
        x: -0.12224,
        y: -1.31017,
        z: 0.28017
    },
    compass: {
        x: -125.02294,
        y: -46.59345,
        z: 41.99643
    },
    dt: 13
}, {
    gyro: {
        x: 19.83322,
        y: -64.06945,
        z: -23.12169
    },
    accel: {
        x: -0.06843,
        y: -1.19367,
        z: 0.32197
    },
    compass: {
        x: -121.31856,
        y: -46.59345,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 11.71108,
        y: -48.98548,
        z: -21.17513
    },
    accel: {
        x: -0.11639,
        y: -1.15329,
        z: 0.24101
    },
    compass: {
        x: -121.31856,
        y: -46.59345,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 11.23017,
        y: -41.82518,
        z: -19.80108
    },
    accel: {
        x: -0.08474,
        y: -1.10997,
        z: 0.13195
    },
    compass: {
        x: -119.46636,
        y: -46.59345,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: 16.1462,
        y: -43.38243,
        z: -20.83925
    },
    accel: {
        x: -0.05723,
        y: -1.06591,
        z: 0.03346
    },
    compass: {
        x: -123.17075,
        y: -48.3855,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 18.16147,
        y: -57.45113,
        z: -22.26673
    },
    accel: {
        x: -0.07305,
        y: -1.04927,
        z: 0.00777
    },
    compass: {
        x: -125.02294,
        y: -46.59345,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 16.56605,
        y: -75.4893,
        z: -31.44231
    },
    accel: {
        x: -0.03604,
        y: -1.08329,
        z: 0.22804
    },
    compass: {
        x: -125.02294,
        y: -46.59345,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 12.89429,
        y: -69.07708,
        z: -31.38887
    },
    accel: {
        x: -0.04164,
        y: -1.28105,
        z: 0.39908
    },
    compass: {
        x: -125.02294,
        y: -46.59345,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 12.00879,
        y: -51.52747,
        z: -24.53391
    },
    accel: {
        x: -0.04895,
        y: -1.28472,
        z: 0.39283
    },
    compass: {
        x: -125.02294,
        y: -48.3855,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: 6.92482,
        y: -38.94731,
        z: -21.12169
    },
    accel: {
        x: -0.05357,
        y: -1.19881,
        z: 0.31909
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 11.87902,
        y: -34.19159,
        z: -17.67895
    },
    accel: {
        x: -0.03604,
        y: -1.09235,
        z: 0.20138
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 16.06986,
        y: -35.01602,
        z: -15.3965
    },
    accel: {
        x: -0.00978,
        y: -1.06004,
        z: 0.09328
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 31.71159
    },
    dt: 12
}, {
    gyro: {
        x: 11.47444,
        y: -44.33663,
        z: -18.54154
    },
    accel: {
        x: 0.01318,
        y: -1.0921,
        z: 0.05244
    },
    compass: {
        x: -128.72732,
        y: -46.59345,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: -10.02937,
        y: -61.13052,
        z: -23.32017
    },
    accel: {
        x: -0.02947,
        y: -1.23405,
        z: 0.19441
    },
    compass: {
        x: -126.87513,
        y: -46.59345,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: -18.12098,
        y: -63.80227,
        z: -15.48047
    },
    accel: {
        x: -0.06794,
        y: -1.34958,
        z: 0.36617
    },
    compass: {
        x: -128.72732,
        y: -46.59345,
        z: 24.85503
    },
    dt: 12
}, {
    gyro: {
        x: -17.54082,
        y: -49.52747,
        z: -3.43467
    },
    accel: {
        x: -0.09083,
        y: -1.29157,
        z: 0.42022
    },
    compass: {
        x: -128.72732,
        y: -46.59345,
        z: 24.85503
    },
    dt: 13
}, {
    gyro: {
        x: -12.64769,
        y: -32.45113,
        z: 6.45846
    },
    accel: {
        x: -0.02703,
        y: -1.1413,
        z: 0.38539
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 21.42675
    },
    dt: 13
}, {
    gyro: {
        x: -4.02937,
        y: -17.99311,
        z: 1.16075
    },
    accel: {
        x: 0.02271,
        y: -1.00155,
        z: 0.30828
    },
    compass: {
        x: -126.87513,
        y: -48.3855,
        z: 26.56917
    },
    dt: 12
}, {
    gyro: {
        x: -2.57899,
        y: -12.63434,
        z: -5.24383
    },
    accel: {
        x: 0.0144,
        y: -0.98588,
        z: 0.11778
    },
    compass: {
        x: -125.02294,
        y: -46.59345,
        z: 26.56917
    },
    dt: 12
}, {
    gyro: {
        x: -7.74693,
        y: -20.558,
        z: -11.98429
    },
    accel: {
        x: -0.00294,
        y: -1.04584,
        z: 0.00873
    },
    compass: {
        x: -125.02294,
        y: -46.59345,
        z: 26.56917
    },
    dt: 12
}, {
    gyro: {
        x: -20.02174,
        y: -34.74884,
        z: -15.10643
    },
    accel: {
        x: -0.03702,
        y: -1.1107,
        z: 0.07982
    },
    compass: {
        x: -125.02294,
        y: -46.59345,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: -17.86144,
        y: -37.23739,
        z: -8.71711
    },
    accel: {
        x: -0.05309,
        y: -1.14668,
        z: 0.18792
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 31.71159
    },
    dt: 12
}, {
    gyro: {
        x: -13.6935,
        y: -27.72594,
        z: -1.1904
    },
    accel: {
        x: -0.07135,
        y: -1.06322,
        z: 0.20402
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 28.28331
    },
    dt: 13
}, {
    gyro: {
        x: -14.07518,
        y: -15.6725,
        z: 1.47373
    },
    accel: {
        x: -0.06843,
        y: -1.00595,
        z: 0.12114
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 28.28331
    },
    dt: 13
}, {
    gyro: {
        x: -20.73166,
        y: -8.99311,
        z: 3.19128
    },
    accel: {
        x: -0.03215,
        y: -0.992,
        z: 0.00368
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 28.28331
    },
    dt: 12
}, {
    gyro: {
        x: -24.99121,
        y: -16.71831,
        z: 0.87831
    },
    accel: {
        x: -0.01173,
        y: -0.89435,
        z: -0.03963
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: -18.66296,
        y: -26.73357,
        z: -0.93849
    },
    accel: {
        x: 0.00732,
        y: -0.89655,
        z: 0.0582
    },
    compass: {
        x: -132.43171,
        y: -43.00933,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: -11.23548,
        y: -27.12289,
        z: -6.24383
    },
    accel: {
        x: 0.04909,
        y: -0.89777,
        z: 0.14588
    },
    compass: {
        x: -132.43171,
        y: -43.00933,
        z: 26.56917
    },
    dt: 12
}, {
    gyro: {
        x: -11.82327,
        y: -20.82518,
        z: -13.1446
    },
    accel: {
        x: 0.042,
        y: -1.05735,
        z: 0.16919
    },
    compass: {
        x: -132.43171,
        y: -43.00933,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: -11.41869,
        y: -16.4206,
        z: -18.48811
    },
    accel: {
        x: 0.09452,
        y: -1.05123,
        z: 0.17231
    },
    compass: {
        x: -126.87513,
        y: -46.59345,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: -15.22785,
        y: -13.83281,
        z: -24.1675
    },
    accel: {
        x: 0.00585,
        y: -1.11584,
        z: 0.1173
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 26.56917
    },
    dt: 12
}, {
    gyro: {
        x: -10.12861,
        y: -16.19922,
        z: -17.86215
    },
    accel: {
        x: -0.01462,
        y: -1.16895,
        z: 0.14252
    },
    compass: {
        x: -130.57951,
        y: -43.00933,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: -6.57899,
        y: -10.6038,
        z: -9.44994
    },
    accel: {
        x: -0.00904,
        y: -1.14595,
        z: 0.16654
    },
    compass: {
        x: -130.57951,
        y: -43.00933,
        z: 29.99745
    },
    dt: 14
}, {
    gyro: {
        x: -7.99121,
        y: -0.76411,
        z: -0.90795
    },
    accel: {
        x: 0.01709,
        y: -1.10507,
        z: 0.12258
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 28.28331
    },
    dt: 12
}, {
    gyro: {
        x: -6.808,
        y: 0.4878,
        z: 4.10731
    },
    accel: {
        x: 0.0144,
        y: -1.05441,
        z: 0.04139
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: -7.62479,
        y: -2.00838,
        z: 3.6493
    },
    accel: {
        x: 0.03981,
        y: -1.01231,
        z: 0.01905
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 29.99745
    },
    dt: 14
}, {
    gyro: {
        x: -8.43395,
        y: -3.31373,
        z: 3.93174
    },
    accel: {
        x: 0.021,
        y: -1.04144,
        z: -0.0021
    },
    compass: {
        x: -126.87513,
        y: -46.59345,
        z: 31.71159
    },
    dt: 12
}, {
    gyro: {
        x: -12.9225,
        y: -7.48166,
        z: 4.34395
    },
    accel: {
        x: 0.04787,
        y: -1.02064,
        z: 0.00896
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: -12.09808,
        y: -12.15342,
        z: 5.09205
    },
    accel: {
        x: 0.02491,
        y: -1.02064,
        z: 0.06613
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: -4.6935,
        y: -10.6954,
        z: 3.09205
    },
    accel: {
        x: 0.0442,
        y: -0.88578,
        z: 0.12859
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 33.42573
    },
    dt: 12
}, {
    gyro: {
        x: 2.17673,
        y: -5.4664,
        z: -2.92322
    },
    accel: {
        x: 0.0381,
        y: -0.88186,
        z: 0.12306
    },
    compass: {
        x: -125.02294,
        y: -46.59345,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 0.71108,
        y: -1.54273,
        z: -7.24383
    },
    accel: {
        x: 0.01953,
        y: -0.92029,
        z: 0.08343
    },
    compass: {
        x: -128.72732,
        y: -46.59345,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: 0.86376,
        y: 0.44963,
        z: -8.77818
    },
    accel: {
        x: 0.01831,
        y: -0.95774,
        z: 0.05988
    },
    compass: {
        x: -130.57951,
        y: -39.42522,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: 2.18437,
        y: 3.5565,
        z: -8.96902
    },
    accel: {
        x: 0.02344,
        y: -0.97364,
        z: 0.04307
    },
    compass: {
        x: -130.57951,
        y: -39.42522,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: -0.28128,
        y: 5.442,
        z: -7.3049
    },
    accel: {
        x: 0.02075,
        y: -0.98686,
        z: 0.01088
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: -0.93014,
        y: 5.21299,
        z: -3.57208
    },
    accel: {
        x: 0.03174,
        y: -1.02626,
        z: -0.01413
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: 3.23017,
        y: 1.61757,
        z: 0.04624
    },
    accel: {
        x: 0.03101,
        y: -1.04535,
        z: -0.05311
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 4.192,
        y: -2.4435,
        z: 5.30579
    },
    accel: {
        x: 0.02149,
        y: -1.07742,
        z: -0.04348
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: -3.42632,
        y: -2.96258,
        z: 10.00808
    },
    accel: {
        x: 0.04103,
        y: -1.0177,
        z: -0.03915
    },
    compass: {
        x: -128.72732,
        y: -37.63317,
        z: 26.56917
    },
    dt: 12
}, {
    gyro: {
        x: -2.14388,
        y: -4.92441,
        z: 0.43556
    },
    accel: {
        x: 0.10038,
        y: -0.82582,
        z: -0.02327
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: -1.65533,
        y: -9.39769,
        z: -9.43467
    },
    accel: {
        x: 0.06399,
        y: -0.84515,
        z: -0.05335
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: 3.97063,
        y: -14.6267,
        z: -17.38887
    },
    accel: {
        x: 0.06789,
        y: -0.89606,
        z: -0.01317
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 31.71159
    },
    dt: 12
}, {
    gyro: {
        x: 1.1004,
        y: -16.87861,
        z: -16.38124
    },
    accel: {
        x: 0.01978,
        y: -1.08035,
        z: -0.0021
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 31.71159
    },
    dt: 12
}, {
    gyro: {
        x: 10.98589,
        y: 71.90002,
        z: -1.97666
    },
    accel: {
        x: -0.10617,
        y: -1.02259,
        z: 0.13676
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: 23.51261,
        y: 14.3046,
        z: 13.28289
    },
    accel: {
        x: 0.06985,
        y: -1.21986,
        z: 0.60711
    },
    compass: {
        x: -130.57951,
        y: -41.21728,
        z: 28.28331
    },
    dt: 13
}, {
    gyro: {
        x: 18.192,
        y: 25.442,
        z: 3.99281
    },
    accel: {
        x: 0.02051,
        y: -0.94207,
        z: 0.00344
    },
    compass: {
        x: -123.17075,
        y: -46.59345,
        z: 29.99745
    },
    dt: 32
}, {
    gyro: {
        x: 28.13093,
        y: 25.43437,
        z: 1.27525
    },
    accel: {
        x: 0.00244,
        y: -0.93816,
        z: 0.05316
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 28.28331
    },
    dt: 24
}, {
    gyro: {
        x: 30.42101,
        y: 28.34276,
        z: -2.48047
    },
    accel: {
        x: 0.04445,
        y: -0.73869,
        z: 0.02721
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: 23.95536,
        y: 35.13666,
        z: -12.90795
    },
    accel: {
        x: 0.03419,
        y: -0.65449,
        z: -0.05335
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 26.56917
    },
    dt: 12
}, {
    gyro: {
        x: -14.8538,
        y: 23.14429,
        z: -15.00719
    },
    accel: {
        x: -0.05844,
        y: -0.94623,
        z: 0.06877
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 26.56917
    },
    dt: 13
}, {
    gyro: {
        x: -32.66296,
        y: 27.29696,
        z: -24.58734
    },
    accel: {
        x: 0.01587,
        y: -1.08133,
        z: 0.59726
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: -40.91487,
        y: 20.06032,
        z: -14.86215
    },
    accel: {
        x: 0.06008,
        y: -1.2436,
        z: 0.01592
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: -19.34998,
        y: 15.67864,
        z: -0.22856
    },
    accel: {
        x: 0.01978,
        y: -1.08623,
        z: -0.0264
    },
    compass: {
        x: -128.72732,
        y: -35.84111,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: -12.95304,
        y: 10.5107,
        z: -4.2133
    },
    accel: {
        x: 0.05397,
        y: -0.99371,
        z: 0.11802
    },
    compass: {
        x: -128.72732,
        y: -35.84111,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: -10.6935,
        y: 10.98398,
        z: 2.48136
    },
    accel: {
        x: 0.04567,
        y: -1.04535,
        z: 0.00849
    },
    compass: {
        x: -128.72732,
        y: -39.42522,
        z: 31.71159
    },
    dt: 12
}, {
    gyro: {
        x: -18.25838,
        y: 14.06795,
        z: -6.87742
    },
    accel: {
        x: 0.10209,
        y: -0.83585,
        z: 0.00151
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: -26.58663,
        y: 16.56414,
        z: -20.2133
    },
    accel: {
        x: 0.07473,
        y: -0.66477,
        z: -0.02423
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: -23.66296,
        y: 27.25879,
        z: -19.2362
    },
    accel: {
        x: -0.01656,
        y: -0.98515,
        z: 0.08511
    },
    compass: {
        x: -128.72732,
        y: -37.63317,
        z: 29.99745
    },
    dt: 12
}, {
    gyro: {
        x: -4.49502,
        y: 39.83131,
        z: -16.20566
    },
    accel: {
        x: -0.00391,
        y: -1.00693,
        z: 0.1358
    },
    compass: {
        x: -128.72732,
        y: -37.63317,
        z: 29.99745
    },
    dt: 13
}, {
    gyro: {
        x: -0.7851,
        y: 34.87711,
        z: -6.83162
    },
    accel: {
        x: -0.01246,
        y: -1.09895,
        z: -0.20374
    },
    compass: {
        x: -130.57951,
        y: -37.63317,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -73.64769,
        y: -16.72594,
        z: 10.22945
    },
    accel: {
        x: -0.00294,
        y: -1.13297,
        z: -0.12289
    },
    compass: {
        x: -128.72732,
        y: -35.84111,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -4.24311,
        y: -13.17632,
        z: 27.01571
    },
    accel: {
        x: 0.03443,
        y: -1.32216,
        z: -0.15923
    },
    compass: {
        x: -126.87513,
        y: -35.84111,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.00116,
        y: 2.3733,
        z: 11.49663
    },
    accel: {
        x: 0.17829,
        y: -0.88088,
        z: 0.13363
    },
    compass: {
        x: -126.87513,
        y: -35.84111,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.22021,
        y: 0.38093,
        z: -0.29727
    },
    accel: {
        x: 0.04518,
        y: -1.00375,
        z: -0.00811
    },
    compass: {
        x: -128.72732,
        y: -37.63317,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: -0.22021,
        y: -0.18395,
        z: -0.24383
    },
    accel: {
        x: 0.0359,
        y: -1.00326,
        z: 0.00968
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -0.25838,
        y: -0.18395,
        z: 0.05388
    },
    accel: {
        x: 0.03981,
        y: -1.00057,
        z: 0.00392
    },
    compass: {
        x: -130.57951,
        y: -39.42522,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.15151,
        y: -0.32136,
        z: 0.33632
    },
    accel: {
        x: 0.0381,
        y: -1.00448,
        z: 0.00896
    },
    compass: {
        x: -130.57951,
        y: -39.42522,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.0546,
        y: -0.14579,
        z: 0.29052
    },
    accel: {
        x: 0.0381,
        y: -1.00522,
        z: 0.00896
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: -0.34235,
        y: -0.30609,
        z: 0.42029
    },
    accel: {
        x: 0.0359,
        y: -1.00252,
        z: 0.01496
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 40.28229
    },
    dt: 13
}, {
    gyro: {
        x: -0.09808,
        y: -0.04655,
        z: 3.86304
    },
    accel: {
        x: -0.05528,
        y: -0.93742,
        z: 0.02625
    },
    compass: {
        x: -128.72732,
        y: -35.84111,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: 0.03169,
        y: 0.3275,
        z: 15.87067
    },
    accel: {
        x: -0.05552,
        y: -0.85005,
        z: 0.00584
    },
    compass: {
        x: -128.72732,
        y: -35.84111,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: 0.03933,
        y: -7.34426,
        z: 15.25235
    },
    accel: {
        x: 0.02564,
        y: -0.74921,
        z: -0.19869
    },
    compass: {
        x: -128.72732,
        y: -37.63317,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: 0.84849,
        y: 6.82368,
        z: 14.21418
    },
    accel: {
        x: 0.0232,
        y: -1.00326,
        z: 0.08703
    },
    compass: {
        x: -128.72732,
        y: -37.63317,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.57899,
        y: 5.31986,
        z: 20.31342
    },
    accel: {
        x: 0.0442,
        y: -1.11951,
        z: 0.36833
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -3.12861,
        y: 4.47253,
        z: 35.84014
    },
    accel: {
        x: -0.04895,
        y: -1.03924,
        z: -0.01365
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.48208,
        y: 4.76261,
        z: 43.50426
    },
    accel: {
        x: -0.01124,
        y: -0.99812,
        z: 0.34095
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.18437,
        y: 4.3962,
        z: 45.96227
    },
    accel: {
        x: 0.02588,
        y: -0.62023,
        z: -0.7158
    },
    compass: {
        x: -128.72732,
        y: -35.84111,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: -1.67823,
        y: -9.28319,
        z: -4.54918
    },
    accel: {
        x: 0.06228,
        y: -0.96606,
        z: -0.14527
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: 0.22253,
        y: 2.72444,
        z: -5.10643
    },
    accel: {
        x: 0.04127,
        y: -1.24752,
        z: -0.35077
    },
    compass: {
        x: -130.57951,
        y: -43.00933,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: -0.35762,
        y: -2.72594,
        z: 0.57296
    },
    accel: {
        x: -0.02996,
        y: -0.99592,
        z: -0.33849
    },
    compass: {
        x: -130.57951,
        y: -43.00933,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: -0.25838,
        y: -0.12289,
        z: 0.00808
    },
    accel: {
        x: 0.00439,
        y: -0.98466,
        z: 0.0313
    },
    compass: {
        x: -123.17075,
        y: -39.42522,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.01411,
        y: -0.36716,
        z: 0.29815
    },
    accel: {
        x: -0.00147,
        y: -1.01599,
        z: -0.00402
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.02174,
        y: 0.11376,
        z: -0.26673
    },
    accel: {
        x: -0.0005,
        y: -1.00155,
        z: 0.0092
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.03701,
        y: -0.70304,
        z: 0.19892
    },
    accel: {
        x: -0.0005,
        y: -1.00571,
        z: 0.01184
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.00116,
        y: -0.19922,
        z: 0.28289
    },
    accel: {
        x: 0.00024,
        y: -1.00252,
        z: 0.01857
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -0.19731,
        y: -0.29082,
        z: 0.13785
    },
    accel: {
        x: -0.00391,
        y: -0.9991,
        z: 0.00705
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.01643,
        y: -0.25266,
        z: 0.20655
    },
    accel: {
        x: -0.00391,
        y: -0.99959,
        z: 0.00729
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.21258,
        y: -0.13052,
        z: 0.27525
    },
    accel: {
        x: -0.00074,
        y: -0.99983,
        z: 0.00825
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 117
}, {
    gyro: {
        x: 0.20727,
        y: -0.22212,
        z: 0.07678
    },
    accel: {
        x: -0.00611,
        y: -0.99983,
        z: 0.00705
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.20727,
        y: -0.02365,
        z: 0.03861
    },
    accel: {
        x: -0.00367,
        y: -0.99934,
        z: 0.01184
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.06223,
        y: -0.24502,
        z: 0.17602
    },
    accel: {
        x: -0.00123,
        y: -0.99298,
        z: 0.01616
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.02937,
        y: -0.07708,
        z: 0.03098
    },
    accel: {
        x: -0.00294,
        y: -1.00057,
        z: 0.00657
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.04696,
        y: -0.26792,
        z: -0.03773
    },
    accel: {
        x: -0.00367,
        y: -1.00595,
        z: 0.01616
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.00116,
        y: -0.19922,
        z: -0.04536
    },
    accel: {
        x: -0.00391,
        y: -1.00399,
        z: 0.01016
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.03933,
        y: 0.02215,
        z: 0.12258
    },
    accel: {
        x: -0.00172,
        y: -1.0013,
        z: 0.01256
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: 0.24544,
        y: -0.26792,
        z: 0.09205
    },
    accel: {
        x: -0.00172,
        y: -1.00057,
        z: 0.01064
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.1462,
        y: -0.14579,
        z: 0.21418
    },
    accel: {
        x: 0.00048,
        y: -1.00791,
        z: 0.01328
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.19963,
        y: -0.12289,
        z: -0.04536
    },
    accel: {
        x: -0.00098,
        y: -0.99934,
        z: 0.0092
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.29655,
        y: -0.28319,
        z: 0.28289
    },
    accel: {
        x: -0.00343,
        y: -1.00057,
        z: 0.00681
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.02174,
        y: -0.09998,
        z: 0.29815
    },
    accel: {
        x: -0.00098,
        y: -1.00399,
        z: 0.01953
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: 0.06223,
        y: -0.13052,
        z: 0.26762
    },
    accel: {
        x: -0.00489,
        y: -1.00301,
        z: 0.0164
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 33.42573
    },
    dt: 12
}, {
    gyro: {
        x: -0.04464,
        y: -0.28319,
        z: 0.25235
    },
    accel: {
        x: -0.00147,
        y: -1.00203,
        z: 0.00753
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: 0.2149,
        y: -0.24502,
        z: 0.19892
    },
    accel: {
        x: -0.00416,
        y: -1.00326,
        z: 0.00944
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 33.42573
    },
    dt: 12
}, {
    gyro: {
        x: -0.09044,
        y: 0.00689,
        z: -0.06063
    },
    accel: {
        x: -0.0044,
        y: -1.00448,
        z: 0.01472
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.0546,
        y: -0.32136,
        z: -0.03009
    },
    accel: {
        x: 0.00073,
        y: -1.00497,
        z: 0.00849
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.25838,
        y: -0.4435,
        z: 0.23708
    },
    accel: {
        x: -0.00172,
        y: -0.99934,
        z: 0.0092
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.22785,
        y: -0.32899,
        z: 0.08441
    },
    accel: {
        x: -0.00147,
        y: -1.00375,
        z: 0.01544
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.28892,
        y: -0.59617,
        z: 0.00808
    },
    accel: {
        x: -0.00709,
        y: -0.9991,
        z: 0.01232
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -0.35762,
        y: -0.11525,
        z: -0.05299
    },
    accel: {
        x: -0.00367,
        y: -1.00301,
        z: 0.00825
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.12098,
        y: -0.34426,
        z: 0.04624
    },
    accel: {
        x: -0.00367,
        y: -1.00473,
        z: 0.01256
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.03701,
        y: -0.17632,
        z: 0.03098
    },
    accel: {
        x: 0.00097,
        y: -1.00106,
        z: 0.01328
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.10571,
        y: -0.03892,
        z: -0.07589
    },
    accel: {
        x: -0.00343,
        y: -1.0035,
        z: 0.01016
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.24311,
        y: -0.19922,
        z: 0.03098
    },
    accel: {
        x: -0.00123,
        y: -1.00155,
        z: 0.01544
    },
    compass: {
        x: -123.17075,
        y: -43.00933,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: -0.21258,
        y: -0.29082,
        z: -0.05299
    },
    accel: {
        x: -0.00391,
        y: -0.99665,
        z: 0.01112
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -0.20495,
        y: -0.28319,
        z: 0.11495
    },
    accel: {
        x: -0.00343,
        y: -1.00228,
        z: 0.02001
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.22021,
        y: -0.35953,
        z: 0.03861
    },
    accel: {
        x: -0.00172,
        y: -1.00228,
        z: 0.0032
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.11334,
        y: -0.32899,
        z: 0.05388
    },
    accel: {
        x: -0.00367,
        y: -1.00057,
        z: 0.01737
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.11566,
        y: -0.03128,
        z: 0.18365
    },
    accel: {
        x: -0.00098,
        y: -1.0013,
        z: 0.0116
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: -0.01411,
        y: -0.38243,
        z: 0.04624
    },
    accel: {
        x: -0.00611,
        y: -0.99934,
        z: 0.00825
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -0.02174,
        y: 0.02979,
        z: 0.10731
    },
    accel: {
        x: -0.00001,
        y: -1.00203,
        z: 0.00944
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.05991,
        y: -0.26029,
        z: 0.17602
    },
    accel: {
        x: -0.0044,
        y: -1.00155,
        z: 0.00992
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.09044,
        y: -0.31373,
        z: 0.16838
    },
    accel: {
        x: -0.00294,
        y: -1.00106,
        z: 0.01016
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.15151,
        y: -0.4206,
        z: 0.08441
    },
    accel: {
        x: -0.00514,
        y: -0.99787,
        z: 0.00681
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.16147,
        y: -0.28319,
        z: 0.19892
    },
    accel: {
        x: -0.00465,
        y: -1.00057,
        z: 0.00777
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 40.28229
    },
    dt: 12
}, {
    gyro: {
        x: 0.06986,
        y: -0.43586,
        z: 0.13785
    },
    accel: {
        x: -0.00391,
        y: -1.00277,
        z: 0.0104
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.06223,
        y: -0.16869,
        z: -0.2591
    },
    accel: {
        x: -0.00514,
        y: -1.00106,
        z: 0.0092
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.12861,
        y: -0.35953,
        z: 0.05388
    },
    accel: {
        x: -0.0044,
        y: -0.99885,
        z: 0.0104
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: 0.10803,
        y: -0.04655,
        z: -0.09879
    },
    accel: {
        x: -0.00147,
        y: -1.00252,
        z: 0.01424
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: 0.06223,
        y: -0.31373,
        z: 0.14548
    },
    accel: {
        x: 0.0017,
        y: -1.00032,
        z: 0.00873
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: 0.02406,
        y: -0.13815,
        z: -0.08353
    },
    accel: {
        x: -0.00318,
        y: -0.9991,
        z: 0.01112
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: -0.06754,
        y: 0.16719,
        z: -0.2133
    },
    accel: {
        x: -0.00562,
        y: -1.00277,
        z: 0.00849
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: 0.22253,
        y: -0.36716,
        z: 0.09968
    },
    accel: {
        x: -0.00074,
        y: -1.00277,
        z: 0.0092
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: -0.09044,
        y: -0.26029,
        z: -0.08353
    },
    accel: {
        x: -0.00343,
        y: -0.99812,
        z: 0.00944
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 35.13987
    },
    dt: 11
}, {
    gyro: {
        x: 0.08513,
        y: -0.06182,
        z: -0.06063
    },
    accel: {
        x: -0.00294,
        y: -1.00155,
        z: 0.01136
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 33.42573
    },
    dt: 12
}, {
    gyro: {
        x: 0.06223,
        y: -0.10762,
        z: -0.01482
    },
    accel: {
        x: -0.00098,
        y: -1.00057,
        z: 0.01016
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 33.42573
    },
    dt: 13
}, {
    gyro: {
        x: -0.18205,
        y: -0.29846,
        z: -0.12933
    },
    accel: {
        x: -0.00245,
        y: -1.00301,
        z: 0.00992
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 33.42573
    },
    dt: 11
}, {
    gyro: {
        x: -0.18205,
        y: -0.10762,
        z: -0.02246
    },
    accel: {
        x: -0.00025,
        y: -0.9991,
        z: 0.01616
    },
    compass: {
        x: -130.57951,
        y: -41.21728,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.03169,
        y: -0.19922,
        z: -0.17513
    },
    accel: {
        x: -0.00318,
        y: -1.0035,
        z: 0.00873
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.13624,
        y: 0.03742,
        z: -0.06826
    },
    accel: {
        x: -0.0044,
        y: -0.99959,
        z: 0.01352
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 35.13987
    },
    dt: 11
}, {
    gyro: {
        x: 0.06986,
        y: -0.06945,
        z: 0.13785
    },
    accel: {
        x: -0.00098,
        y: -1.00522,
        z: 0.01088
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.15915,
        y: -0.29846,
        z: 0.03861
    },
    accel: {
        x: -0.00391,
        y: -1.00106,
        z: 0.01472
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: -0.15151,
        y: -0.38243,
        z: 0.12258
    },
    accel: {
        x: 0.00195,
        y: -0.99983,
        z: 0.00992
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.20495,
        y: -0.09998,
        z: -0.04536
    },
    accel: {
        x: -0.00318,
        y: -1.00252,
        z: 0.01016
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.04696,
        y: -0.09998,
        z: 0.13785
    },
    accel: {
        x: -0.00343,
        y: -1.0013,
        z: 0.01376
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.08513,
        y: -0.35953,
        z: 0.07678
    },
    accel: {
        x: -0.00221,
        y: -1.00301,
        z: 0.0116
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 11
}, {
    gyro: {
        x: 0.09276,
        y: -0.09998,
        z: 0.03861
    },
    accel: {
        x: -0.00538,
        y: -1.00301,
        z: 0.014
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.02174,
        y: -0.39769,
        z: 0.06151
    },
    accel: {
        x: -0.00172,
        y: -0.99934,
        z: 0.00705
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: 0.20727,
        y: -0.17632,
        z: 0.09205
    },
    accel: {
        x: -0.00123,
        y: -1.00155,
        z: 0.0044
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.06986,
        y: -0.25266,
        z: 0.32105
    },
    accel: {
        x: -0.00172,
        y: -1.0035,
        z: 0.00944
    },
    compass: {
        x: -126.87513,
        y: -41.21728,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.03701,
        y: -0.25266,
        z: 0.00044
    },
    accel: {
        x: -0.00074,
        y: -1.00155,
        z: 0.014
    },
    compass: {
        x: -123.17075,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.00647,
        y: -0.09998,
        z: -0.07589
    },
    accel: {
        x: -0.00367,
        y: -0.9991,
        z: 0.01232
    },
    compass: {
        x: -123.17075,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.1233,
        y: -0.29846,
        z: 0.13785
    },
    accel: {
        x: -0.00367,
        y: -1.0035,
        z: 0.01232
    },
    compass: {
        x: -123.17075,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.06754,
        y: -0.37479,
        z: -0.10643
    },
    accel: {
        x: -0.00465,
        y: -0.99959,
        z: 0.00584
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.03169,
        y: -0.14579,
        z: 0.02334
    },
    accel: {
        x: 0.00048,
        y: -0.99885,
        z: 0.01328
    },
    compass: {
        x: -128.72732,
        y: -41.21728,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.09044,
        y: -0.08472,
        z: 0.13021
    },
    accel: {
        x: -0.00269,
        y: -1.00228,
        z: 0.01496
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.02174,
        y: -0.29082,
        z: 0.09968
    },
    accel: {
        x: -0.0044,
        y: -0.99518,
        z: 0.00584
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.04696,
        y: -0.07708,
        z: -0.03773
    },
    accel: {
        x: -0.00514,
        y: -1.00106,
        z: 0.01424
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.1004,
        y: -0.12289,
        z: 0.17602
    },
    accel: {
        x: -0.00123,
        y: -1.00326,
        z: 0.01184
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.12861,
        y: -0.14579,
        z: -0.07589
    },
    accel: {
        x: -0.00343,
        y: -1.00179,
        z: 0.01304
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: -0.19731,
        y: -0.04655,
        z: -0.07589
    },
    accel: {
        x: -0.00269,
        y: -1.0013,
        z: 0.0116
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 35.13987
    },
    dt: 13
}, {
    gyro: {
        x: 0.01643,
        y: -0.28319,
        z: -0.12933
    },
    accel: {
        x: -0.00489,
        y: -1.00448,
        z: 0.0128
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 35.13987
    },
    dt: 11
}, {
    gyro: {
        x: 0.1462,
        y: -0.26792,
        z: 0.04624
    },
    accel: {
        x: -0.00196,
        y: -0.99641,
        z: 0.01232
    },
    compass: {
        x: -128.72732,
        y: -44.80139,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.19731,
        y: -0.22212,
        z: 0.03098
    },
    accel: {
        x: -0.0044,
        y: -0.99812,
        z: 0.01208
    },
    compass: {
        x: -123.17075,
        y: -39.42522,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: -0.22785,
        y: -0.19922,
        z: 0.01571
    },
    accel: {
        x: -0.00514,
        y: -1.0035,
        z: 0.01328
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 33.42573
    },
    dt: 11
}, {
    gyro: {
        x: 0.08513,
        y: -0.22212,
        z: 0.10731
    },
    accel: {
        x: -0.00074,
        y: -1.00326,
        z: 0.01304
    },
    compass: {
        x: -125.02294,
        y: -43.00933,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: 0.03933,
        y: -0.34426,
        z: -0.17513
    },
    accel: {
        x: -0.00172,
        y: -1.00203,
        z: 0.01688
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 40.28229
    },
    dt: 13
}, {
    gyro: {
        x: 0.1004,
        y: -0.20685,
        z: 0.11495
    },
    accel: {
        x: -0.0005,
        y: -1.00399,
        z: 0.0116
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 40.28229
    },
    dt: 12
}, {
    gyro: {
        x: 0.04696,
        y: 0.01452,
        z: 0.36685
    },
    accel: {
        x: -0.00318,
        y: -0.99934,
        z: 0.00968
    },
    compass: {
        x: -123.17075,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.06754,
        y: -0.03892,
        z: 0.00044
    },
    accel: {
        x: -0.00416,
        y: -0.99983,
        z: 0.01833
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 33.42573
    },
    dt: 11
}, {
    gyro: {
        x: 0.23017,
        y: -0.19922,
        z: 0.03098
    },
    accel: {
        x: -0.00123,
        y: -1.00277,
        z: 0.00873
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: 0.00879,
        y: -0.01602,
        z: -0.07589
    },
    accel: {
        x: -0.00245,
        y: -0.99812,
        z: 0.00873
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 40.28229
    },
    dt: 13
}, {
    gyro: {
        x: -0.24311,
        y: -0.45876,
        z: 0.21418
    },
    accel: {
        x: -0.00391,
        y: -1.00497,
        z: 0.01232
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 40.28229
    },
    dt: 11
}, {
    gyro: {
        x: -0.18205,
        y: -0.17632,
        z: -0.07589
    },
    accel: {
        x: -0.00098,
        y: -1.00277,
        z: 0.00753
    },
    compass: {
        x: -126.87513,
        y: -44.80139,
        z: 40.28229
    },
    dt: 12
}, {
    gyro: {
        x: -0.01411,
        y: -0.25266,
        z: -0.07589
    },
    accel: {
        x: -0.00391,
        y: -1.00203,
        z: 0.00681
    },
    compass: {
        x: -123.17075,
        y: -39.42522,
        z: 31.71159
    },
    dt: 13
}, {
    gyro: {
        x: -0.08281,
        y: 0.00689,
        z: 0.02334
    },
    accel: {
        x: -0.00587,
        y: -1.00203,
        z: 0.0116
    },
    compass: {
        x: -125.02294,
        y: -39.42522,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: 0.00116,
        y: -0.40533,
        z: -0.12169
    },
    accel: {
        x: -0.00221,
        y: -1.00326,
        z: 0.01088
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.12861,
        y: -0.16105,
        z: 0.09205
    },
    accel: {
        x: -0.00636,
        y: -1.00106,
        z: 0.0164
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 11
}, {
    gyro: {
        x: 0.1233,
        y: -0.21449,
        z: 0.15311
    },
    accel: {
        x: -0.00416,
        y: -1.0013,
        z: 0.00657
    },
    compass: {
        x: -125.02294,
        y: -41.21728,
        z: 36.85401
    },
    dt: 13
}, {
    gyro: {
        x: -0.25075,
        y: -0.32136,
        z: -0.09116
    },
    accel: {
        x: -0.0044,
        y: -1.00057,
        z: 0.01496
    },
    compass: {
        x: -128.72732,
        y: -43.00933,
        z: 36.85401
    },
    dt: 11
}, {
    gyro: {
        x: -0.16678,
        y: -0.38243,
        z: 0.07678
    },
    accel: {
        x: 0.00121,
        y: -1.00522,
        z: 0.02001
    },
    compass: {
        x: -123.17075,
        y: -44.80139,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.03169,
        y: -0.09235,
        z: 0.15311
    },
    accel: {
        x: -0.00562,
        y: -1.00057,
        z: 0.01208
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 40.28229
    },
    dt: 13
}, {
    gyro: {
        x: -0.13624,
        y: -0.29846,
        z: 0.09968
    },
    accel: {
        x: -0.00221,
        y: -1.00326,
        z: 0.00896
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 40.28229
    },
    dt: 11
}, {
    gyro: {
        x: -0.09808,
        y: -0.19922,
        z: 0.03861
    },
    accel: {
        x: -0.00782,
        y: -1.00106,
        z: 0.01448
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 40.28229
    },
    dt: 12
}, {
    gyro: {
        x: -0.31182,
        y: 0.06032,
        z: 0.06151
    },
    accel: {
        x: -0.0005,
        y: -0.99959,
        z: 0.0104
    },
    compass: {
        x: -125.02294,
        y: -44.80139,
        z: 40.28229
    },
    dt: 12
}, {
    gyro: {
        x: -0.13624,
        y: -0.13052,
        z: 0.15311
    },
    accel: {
        x: -0.00147,
        y: -0.99959,
        z: 0.01256
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 33.42573
    },
    dt: 12
}, {
    gyro: {
        x: -0.15151,
        y: -0.09998,
        z: -0.04536
    },
    accel: {
        x: 0.00121,
        y: -1.00106,
        z: 0.01857
    },
    compass: {
        x: -126.87513,
        y: -43.00933,
        z: 35.13987
    },
    dt: 12
}, {
    gyro: {
        x: -0.13624,
        y: -0.11525,
        z: -0.13696
    },
    accel: {
        x: -0.00758,
        y: -0.99861,
        z: 0.01568
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 38.56815
    },
    dt: 13
}, {
    gyro: {
        x: 0.34467,
        y: -0.23739,
        z: 0.06915
    },
    accel: {
        x: -0.0044,
        y: -1.00179,
        z: 0.01064
    },
    compass: {
        x: -126.87513,
        y: -39.42522,
        z: 38.56815
    },
    dt: 12
}, {
    gyro: {
        x: -0.12861,
        y: -0.26029,
        z: -0.05299
    },
    accel: {
        x: -0.00367,
        y: -1.00228,
        z: 0.01088
    },
    compass: {
        x: -125.02294,
        y: -37.63317,
        z: 36.85401
    },
    dt: 12
}, {
    gyro: {
        x: 0.00879,
        y: -0.61144,
        z: -0.08353
    },
    accel: {
        x: -0.00465,
        y: -1.0035,
        z: 0.0092
    },
    compass: {
        x: -126.87513,
        y: -37.63317,
        z: 36.85401
    },
    dt: 12
}];


// [INFO][6:42 24/11/2016]:Initialization MPU9250 ....
// [INFO][6:42 24/11/2016]:Reset configuration MPU9250.
// [INFO][6:42 24/11/2016]:Enabled magnetometer. Starting initialization ....
// [INFO][6:42 24/11/2016]:END of magnetometer initialization.
// [INFO][6:42 24/11/2016]:END of MPU9150 initialization.
// [INFO][6:42 24/11/2016]:MPU9250:
// [INFO][6:42 24/11/2016]:--> Device address: 0x68
// [INFO][6:42 24/11/2016]:--> i2c bus: /dev/i2c-2
// [INFO][6:42 24/11/2016]:--> Device ID: 0x71
// [INFO][6:42 24/11/2016]:--> BYPASS enabled: Yes
// [INFO][6:42 24/11/2016]:--> SleepEnabled Mode: Off
// [INFO][6:42 24/11/2016]:--> i2c Master Mode: Disabled
// [INFO][6:42 24/11/2016]:--> Power Management (0x6B, 0x6C):
// [INFO][6:42 24/11/2016]:  --> Clock Source: 1 (Auto selects the best available clock source)
// [INFO][6:42 24/11/2016]:  --> Accel enabled (x, y, z): (Yes, Yes, Yes)
// [INFO][6:42 24/11/2016]:  --> Gyro enabled (x, y, z): (Yes, Yes, Yes)
// [INFO][6:42 24/11/2016]:Accelerometer:
// [INFO][6:42 24/11/2016]:--> Full Scale Range (0x1C): 2g (0)
// [INFO][6:42 24/11/2016]:--> Scalar: 1/16384
// [INFO][6:42 24/11/2016]:--> Calibration:
// [INFO][6:42 24/11/2016]:  --> Offset:
// [INFO][6:42 24/11/2016]:    --> x: 0.013922932942708334
// [INFO][6:42 24/11/2016]:    --> y: 0.0071199544270833335
// [INFO][6:42 24/11/2016]:    --> z: -0.009103597005208333
// [INFO][6:42 24/11/2016]:  --> Scale:
// [INFO][6:42 24/11/2016]:    --> x: -0.9887548828125,1.01349365234375
// [INFO][6:42 24/11/2016]:    --> y: -0.9904020182291666,1.0107552083333333
// [INFO][6:42 24/11/2016]:    --> z: -1.0237027994791668,1.0072054036458333
// [INFO][6:42 24/11/2016]:Gyroscope:
// [INFO][6:42 24/11/2016]:--> Full Scale Range (0x1B): +250dps (0)
// [INFO][6:42 24/11/2016]:--> Scalar: 1/131
// [INFO][6:42 24/11/2016]:--> Bias Offset:
// [INFO][6:42 24/11/2016]:  --> x: 0.3217709923664121
// [INFO][6:42 24/11/2016]:  --> y: -1.1915877862595428
// [INFO][6:42 24/11/2016]:  --> z: -0.08352671755725186
// [INFO][6:42 24/11/2016]:Magnetometer (Compass):
// [INFO][6:42 24/11/2016]:--> i2c address: 0xc
// [INFO][6:42 24/11/2016]:--> Device ID: 0x48
// [INFO][6:42 24/11/2016]:--> Mode: 0x06 (Continuous measurement mode 2: 100Hz)
// [INFO][6:42 24/11/2016]:--> Scalars:
// [INFO][6:42 24/11/2016]:  --> x: 1.19921875
// [INFO][6:42 24/11/2016]:  --> y: 1.2109375
// [INFO][6:42 24/11/2016]:  --> z: 1.16015625
